---
title: "Fetal Brain - Histone modifications"
author: "Gloria Li"
date: "January 27, 2014"
output:
  html_document:
    keep_md: yes
    toc: yes
---

Update `r date()`

```{r include = FALSE}
library(ggplot2)
library(stringr)
library(dplyr)
library(reshape2)
library(gridExtra)
library(knitr)
load("/projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/FetalBrain_FindER.Rdata")
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE, fig.width = 8)
```

## Sanity check   

* No. of peaks, No. of enriched bases, and average peak length seem reasonable except for the unusual high No. of peaks in GE HuFNSC04 input library.              

```{r summary, fig.height = 10, fig.width=8}
(FindER_summary_figure)
```

## Correlation with protein-coding gene RPKM 

* Overlapping H3K4me3 and H3K27me3 FindER peaks with protein-coding gene promoters (TSS +/- 1500bp), and overlapping H3K36me3 with genebody.    
* Overall correlations are as expected, with H3K4m3 and H3K36me3 marked genes showing higher RPKM, and H3K27me3 and bivalent promoter marked showing lower RPKM.   

```{r RPKM, fig.height = 6, fig.width = 8}
(HisMod_RPKM_figure)
```

## Differentially marked genes

* Calculate H3K4me3 and H3K27me3 signal from wig at promoters (TSS +/- 1500bp) and rank all genes (pc + nc).    
* Differential marked genes are defined as rank difference >= 5000.           
* For H3K27me3 differential marked genes, there are opposite trends in pc and nc, _why?_           

### MZ twins

```{r DM_MZ1, fig.height = 6}
(H3K4me3_H3K27me3_promoter_drank_MZ_figure)
(DM_H3K4me3_Brain_Subject1_DAVID)
(DM_H3K4me3_Brain_Subject2_DAVID)
```
```{r DM_MZ2, fig.height=2}
(DM_H3K27me3_Brain_Subject1_DAVID)
(DM_H3K27me3_Brain_Subject2_DAVID)
(DM_H3K27me3_Cortex_Subject1_DAVID)
(DM_H3K27me3_Cortex_Subject2_DAVID)
```
```{r DM_MZ3, fig.height=12}
(DM_H3K27me3_GE_Subject1_DAVID)
```
```{r DM_MZ4, fig.height=5}
(DM_H3K27me3_GE_Subject2_DAVID)
grid.arrange(gTree(children = venn_Brain01_Brain02DE_epi), gTree(children = venn_Cortex01_Cortex02DE_epi), gTree(children = venn_GE01_GE02DE_epi), nrow = 1)
```

* Brain01 vs Brain02: DE genes that are DM in both H3K4me3 and H3K27me3         
```{r DM_MZ5, results='asis'}
kable(Brain01_Brain02DE_epi_H3K4me3_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

* Brain01 vs Brain02: DE genes that are DM in both DMR and H3K27me3         

```{r DM_MZ6, results='asis'}
kable(Brain01_Brain02DE_epi_DMR_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

* Cortex01 vs Cortex02: DE genes that are DM in both DMR and H3K27me3         

```{r DM_MZ7, results='asis'}
kable(Cortex01_Cortex02DE_epi_DMR_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

* GE01 vs GE02: DE genes that are DM in both DMR and H3K27me3         

```{r DM_MZ8, results='asis'}
kable(GE01_GE02DE_epi_DMR_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

### NPCs

```{r DM_NPC1, fig.height=6}
(H3K4me3_H3K27me3_promoter_drank_NPC_figure)
(DM_H3K4me3_NPC_Subject2_Cortex_DAVID)
(DM_H3K4me3_NPC_Subject2_GE_DAVID)
```
```{r DM_NPC2, fig.height=2}
(DM_H3K27me3_NPC_Subject1_Cortex_DAVID)
```
```{r DM_NPC3, fig.height=6}
(DM_H3K27me3_NPC_Subject1_GE_DAVID)
(DM_H3K27me3_NPC_Subject2_Cortex_DAVID)
(DM_H3K27me3_NPC_Subject2_GE_DAVID)
grid.arrange(gTree(children = venn_Cortex01_GE01DE_epi), gTree(children = venn_Cortex02_GE02DE_epi), nrow = 1)
(DM_H3K27me3_DE_NPC_Subject1_DAVID)
(DM_H3K4me3_DE_NPC_Subject2_DAVID)
(DM_H3K27me3_DE_NPC_Subject2_DAVID)
```

* Cortex01 vs GE01: DE genes that are DM in both DMR and H3K7me3        

```{r DM_NPC4, results='asis'}
kable(Cortex01_GE01DE_epi_DMR_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

* Cortex02 vs GE02: DE genes that are DM in DMR, H3K4me3, and H3K7me3        

```{r DM_NPC5, results='asis'}
kable(Cortex02_GE02DE_epi_DMR_H3K4me3_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

### GW

```{r DM_GW1, fig.height=6}
(H3K4me3_H3K27me3_promoter_drank_GW_figure)
```
```{r DM_GW2, fig.height=8}
(DM_H3K4me3_GW_GE_GW13_DAVID)
(DM_H3K4me3_GW_GE_GW17_DAVID)
(DM_H3K27me3_GW_GE_GW13_DAVID)
(DM_H3K27me3_GW_GE_GW17_DAVID)
grid.newpage()
grid.draw(venn_GE02_GE04DE_epi)
```
```{r DM_GW3, fig.height=2}
(DM_H3K4me3_DE_GW_GE_DAVID)
```
```{r DM_GW4, fig.height=5}
(DM_H3K27me3_DE_GW_GE_DAVID)
```

* GE02 vs GE04: DE genes that are DM in DMR, H3K4me3, and H3K27me3      

```{r DM_GW5, results='asis'}
kable(GE02_GE04DE_epi_DMR_H3K4me3_H3K27me3%>%select(name, description), format = "html", align = c("r", "r"))
```

## Core enhancers 

* Overlapping all NPC enhancers (Cortex01, Cortex02, GE01, GE02, GE04), in total `r system("wc -l /projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/H3K4me1/core/core_enhancers.bed | awk '{print $1}'", intern = T)` regions, average length `r system("less /projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/H3K4me1/core/core_enhancers.bed | awk '{s=s+$3-$2; c=c+1}END{print s/c}'", intern = T)`bp.     

### GWAS in core enhancers  

* `r length(unique(GWAS_core_enhancer$enhancerID))` enhancers overlap with GWAS, involved in `r nrow(GWAS_core_enhancer_trait)` traits.     
* Several traits are related to brain development and function: Cortical structure, Glaucoma (exfoliation), Glioblastoma, Neuranatomic and neurocognitive phenotypes, Neuroblastoma (high-risk), Odorant perception (isobutyraldehyde), Schizophrenia (cytomegalovirus infection interaction), Alzheimer's disease biomarkers.        

### Homer TFBSs

* There are `r nrow(homer_core_enhancer_top)` TFs significantly (Benjamini q value < 0.01) enriched in core enhancers and present in > 20% of the core enhancers.    
* Ptf1a: plays an important role in cerebellar and pancreatic development.       
* Isl1: central to the development of pancreatic cell lineages and may also be required for motor neuron generation.        
* Lhx3: involved in the development of interneurons and motor neurons.    
* NF1: negative regulator of the ras signal transduction pathway, associated with neurofibromatosis type 1.     
* Sox3: function as a switch in neuronal development. Keeps neural cells undifferentiated by counteracting the activity of proneural proteins and suppresses neuronal differentiation.       
* Olig2: oligodendrocyte specific marker.      

```{r core_enhancer_homer}
(homer_core_enhancer_figure)
```

### Overlapping with WGBS UMRs

* UMRs between neurospheres (cortex vs GE) are enriched in enhancers (H3K4me1 enriched regions).      
* Between gestational weeks, GW13 UMRs are enriched in enhancers, but GW17 UMRs are not.      
* UMRs overlaped with enhancers are highly enriched for brain development terms. For comparing between neurospheres, GE enhancer UMRs in HuFNSC04 have no significant enrichment. And for comparing between gestational weeks, GW13 enhancer UMRs in Cortex also have no enriched terms.       

```{r core_enhancer_UMR, fig.height = 6, fig.width = 8}
(UMR_enhancer_summary_figure)
```
```{r core_enhancer_UMR_GREAT, fig.height = 15, fig.width = 8}
(GREAT_Cortex_UMR_enhancer_HuFNSC02)
(GREAT_GE_UMR_enhancer_HuFNSC02)
(GREAT_Cortex_UMR_enhancer_HuFNSC04)
(GREAT_GW17_UMR_enhancer_Cortex)
(GREAT_GW17_UMR_enhancer_GE)
(GREAT_GW13_UMR_enhancer_GE)
```

## Unique enhancers 

* Overall, there are about 20-40% enhancers that are unique within each pairwise comparisons.      
* Between MZ twins, HuFNSC01 has more unique enhancers than HuFNSC02 in all three cell types. The asymmetry between two progenitor cell types and between GWs is not supported by both samples.   
* The intersect of two samples between both progenitors and GWs are statistically significant.    

```{r unique_enhancer, results='asis', fig.height = 8}
kable(unique_enhancers_summary, format = "html", align = c("l", "l", "c", "c", "c", "c"))
grid.arrange(gTree(children = venn_unique_enhancer_Cortex), gTree(children = venn_unique_enhancer_GE), 
             gTree(children = venn_unique_enhancer_GW13), gTree(children = venn_unique_enhancer_GW17), nrow = 2)
grid.text("Neurospheres unique enhancers\n in Cortex", 0.25, 0.95)
grid.text("Neurospheres unique enhancers\n in GE", 0.75, 0.95)
grid.text("GW unique enhancers\n in GW13", 0.25, 0.5)
grid.text("GW unique enhancers\n in GW17", 0.75, 0.5)
```

### Functional enrichment of unique enhancers

* For MZ twins, no GREAT enriched terms in cortex and GE. In mixed brain, HuFNSC02 unique enhancers show brain organ development terms, WNT pathway, and kidney-related terms.      
* For cortex and GE progenitors, both showed brain-related terms, more so in cortex.     
* For GWs, both GW13 and GW17 unique enhancers showed enrichment for brain organ development terms. GW17 also showed WNT pathway terms.      

```{r unique_enhancer_GREAT1, fig.height=12}
(GREAT_unique_enhancer_HuFNSC01)
```
```{r unique_enhancer_GREAT2, fig.height=8}
(GREAT_unique_enhancer_HuFNSC02)
```
```{r unique_enhancer_GREAT3, fig.height=6}
(GREAT_unique_enhancer_Cortex)
```
```{r unique_enhancer_GREAT4, fig.height=4}
(GREAT_unique_enhancer_GE)
(GREAT_unique_enhancer_GW13)
(GREAT_unique_enhancer_GW17)
```

### GWAS in unique enhancers 

* All sets of unique enhancers showed brain or brain disease related GWAS sites, such as Gliomas, Alzheimer's disease, Autism, Schizophrenia or bipolar disorder, Cognitive performance, Neuroblastoma (high-risk), Normalized brain volume, Intelligence.               

### Homer TFBSs
#### Unique enhancers between GW13 and GW17 

* Intersect of unique enhancers between GE01 vs GE04 and GE02 vs GE04. There are `r system("wc -l /projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/H3K4me1/unique/unique_enhancer.GW.GW13.bed | awk '{print $1}'", intern = T)` GW13-specific enhancers, and `r system("wc -l /projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/H3K4me1/unique/unique_enhancer.GW.GW17.bed | awk '{print $1}'", intern = T)` GW17-specific enhancers.          
* There are `r nrow(homer_unique_enhancer_GW_GW13)` TFs significantly (Benjamini q-value < 0.01) enriched in GW13-specific enhancers, and `r nrow(homer_unique_enhancer_GW_GW17)` in GW17-specific enhancers. Among them, `r nrow(homer_unique_enhancer_GW_common)` TFs are in common.        
* In the common TFs, `r nrow(homer_unique_enhancer_GW_common %>% filter(GW13_Percent_Target >= 20 | GW17_Percent_Target >= 20))` are present in > 20% enhancers in either GW13 or GW17. Their percent occupancy in GW13 and GW17 specific enhancers are similar, and downstream target genes are significantly overlapped (for example, Sox3 hypergeometric test p-value = 0).      
* There are only `r nrow(homer_unique_enhancer_GW_GW13only)` GW13-specific TFs, and they all present in < 5% of the enhancers.    
* There are `r nrow(homer_unique_enhancer_GW_GW17only)` GW17-specific TFs, and `r nrow(filter(homer_unique_enhancer_GW_GW17only, Percent.of.Target.Sequences.with.Motif >= 20))` of them are present in > 20% of the enhancers, including Olig2.      
* GW17-specific enhancers overlapped with Olig2 binding sites are associated with `r nrow(homer_unique_enhancer_GW_GW17only_Olig2_DE_cortex)` GW-specific DE genes in cortex, and `r nrow(homer_unique_enhancer_GW_GW17only_Olig2_DE_GE)` in GE. Both gene lists are enriched for neuron development terms.              

```{r unique_enhancer_homer_GW}
(homer_unique_enhancer_GW_common_figure + ggtitle("Common TFs in GW unique enhancers"))
plot.new()
grid.draw(Venn_homer_unique_enhancer_GW_common_Sox3)
(homer_unique_enhancer_GW_GW13only_figure + ggtitle("GW13-specific TFs in GW unique enhancers"))
(homer_unique_enhancer_GW_GW17only_figure + ggtitle("GW17-specific TFs in GW unique enhancers"))
plot.new()
grid.draw(Venn_homer_unique_enhancer_GW_GW17only_Olig2)
(homer_unique_enhancer_GW_GW17only_Olig2_DE_cortex_DAVID)
(homer_unique_enhancer_GW_GW17only_Olig2_DE_GE_DAVID)
```

* Olig2 downstream DE genes in cortex (distance to TSS < 10kb)    
  + EMX2: Empty Spiracles Homeobox 2. acts to generate the boundary between the roof and archipallium in the developing brain. May function in combinations with OTX1/2 to specify cell fates in the developing central nervous system.       
  + RGS10: Regulator Of G-Protein Signaling 10. associated with schizophrenia.     
  + LGI1: Leucine-Rich, Glioma Inactivated 1. This gene is predominantly expressed in neural tissues and its expression is reduced in low grade brain tumors and significantly reduced or absent in malignant gliomas. Mutations in this gene result in autosomal dominant lateral temporal epilepsy. May play a role in the control of neuroblastoma cell survival.     
  
![Olig2_cortex_DE](./HisMod_files/figure-html/homer_unique_enhancer_GW_GW17only_Olig2_DE_cortex_network.png)          

* Olig2 downstream DE genes in GE (distance to TSS < 10kb)      
  + PAX6: Paired Box 6. expressed in the developing nervous system, and in developing eyes. Mutations in this gene are known to cause ocular disorders such as aniridia and Peter's anomaly.     
  + HIVEP2: Human Immunodeficiency Virus Type I Enhancer Binding Protein 2. zinc finger-containing transcription factors. The encoded protein regulates transcription by binding to regulatory regions of various cellular and viral genes that maybe involved in growth, development and metastasis.    
  
![Olig2_GE_DE](./HisMod_files/figure-html/homer_unique_enhancer_GW_GW17only_Olig2_DE_GE_network.png)       

#### Unique enhancers between cortex and GE

* Intersect of unique enhancers between cortex01 vs GE01 and cortex02 vs GE02. There are `r system("wc -l /projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/H3K4me1/unique/unique_enhancer.Neurospheres.Cortex.bed | awk '{print $1}'", intern = T)` cortex-specific enhancers, and `r system("wc -l /projects/epigenomics/users/lli/FetalBrain/ChIPseq/ER/H3K4me1/unique/unique_enhancer.Neurospheres.GE.bed | awk '{print $1}'", intern = T)` GE-specific enhancers.     
* There are `r nrow(homer_unique_enhancer_Neurospheres_Cortex)` TFs significantly (Benjamini q-value < 0.01) enriched in cortex-specific enhancers, and `r nrow(homer_unique_enhancer_Neurospheres_GE)` in GE-specific enhancers. Among them, `r nrow(homer_unique_enhancer_Neurospheres_common)` TFs are in common.        
* In the common TFs, `r nrow(homer_unique_enhancer_Neurospheres_common %>% filter(Cortex_Percent_Target >= 20 | GE_Percent_Target >= 20))` are present in > 20% enhancers in either cortex or GE. Their percent occupancy in cortex and GE specific enhancers are similar, and downstream target genes are significantly overlapped (for example, Lhx3 hypergeometric test p-value = 0).      
* There are only `r nrow(homer_unique_enhancer_Neurospheres_CortexOnly)` cortex-specific TFs, and they all present in < 20% of the enhancers.    
* There are `r nrow(homer_unique_enhancer_Neurospheres_GEonly)` GW17-specific TFs, and `r nrow(filter(homer_unique_enhancer_Neurospheres_GEonly, Percent.of.Target.Sequences.with.Motif >= 20))` of them are present in > 20% of the enhancers, including Olig2.      
* `r nrow(homer_unique_enhancer_Neurospheres_GEonly_Olig2_DE)` GE-specific enhancers overlapped with Olig2 binding sites are associated with NPC-specific DE genes, and the DE genes are enriched for neuron development terms.              

```{r unique_enhancer_homer_neurospheres}
(homer_unique_enhancer_Neurospheres_common_figure + ggtitle("Common TFs in NPC unique enhancers"))
plot.new()
grid.draw(Venn_homer_unique_enhancer_Neurospheres_common_Lhx3)
(homer_unique_enhancer_Neurospheres_CortexOnly_figure + ggtitle("cortex-specific TFs in NPC unique enhancers"))
(homer_unique_enhancer_Neurospheres_GEonly_figure + ggtitle("GE-specific TFs in NPC unique enhancers"))
plot.new()
grid.draw(Venn_homer_unique_enhancer_Neurospheres_GEonly_Olig2)
(homer_unique_enhancer_Neurospheres_GEonly_Olig2_DE_DAVID)
```


