---
output:
  html_document:
    keep_md: true
    toc: yes
---
Fetal Brain isoform analysis 
========================================================

Gloria Li         
`r date()` 

<!-- re-knit after modify isoform.std.R script -->

```{r include = FALSE}
library(knitr)
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

```{r data}
setwd("~/快盘/FetalBrain/RNAseq/isoform/")
load("FetalBrain_isoform.Rdata")
library(ggplot2)
library(xtable)
library(VennDiagram)
```

## Isoform identification
  * DEfine on exons: FDR = 0.01     
  * Exon expressed in one sample ($\ge$ 10% gene RPKM) and not expressed in the other ($\le$ 1% gene RPKM)   
  * Gene is not DE: DEfine FDR = 0.01
  * Gene is expressed in both samples: gene RPKM > 0.01         

## Isoform genes between cortex and GE

  * On average, __`r round(mean(cortex_GE_summary[, ncol(cortex_GE_summary)]), 0)`__ genes are identified as isoforms between cortex and GE in each individual. __`r length(isoform_cortex_GE_gene_dup)`__ genes are shared in at least two individuals.       
  * There are more individual-specific isoforms than found in breast cells, although the overlap across individuals are still significant.     

```{r cortex_ge, results='asis'}
cortex_GE_summary <- xtable(cortex_GE_summary)
align(cortex_GE_summary) <- "lcccccc"
print(cortex_GE_summary, type = "html", include.rownames = T)

venn_cortex_GE <- venn.diagram(isoform_cortex_GE, filename = NULL, fill = c("red", "blue", "green", "yellow"), main = "Venn diagram of cortex vs GE isoforms", main.cex = 2)
plot.new()
grid.draw(venn_cortex_GE)
```

### Functional enrichment analysis 
  * Individual specific isoforms between cortex and GE have __no__ significantly enriched terms, suggesting they are more likely random events without biological functions.          
  * Isoforms shared by at least two individuals are enriched in terms related to __cellular signaling__. InterPro protein domain enrichment show enriched terms similar to those observed in breast isoforms.         

```{r enrich_cortex_ge, fig.width=10, fig.height = 8}
(isoform_cortex_GE_enrich)
```

## Isoform genes between individuals

  * On average, __`r round(mean(individual_summary[1:3, ncol(individual_summary)]), 0)`__ genes are identified as isoforms between HuFNSC01 and HuFNSC02 in each cell type. __`r length(isoform_HuFNSC01_HuFNSC02_gene_shared)`__ genes are shared by all three cell types.              
  * On average, __`r round(mean(individual_summary[4:5, ncol(individual_summary)]), 0)`__ genes are identified as isoforms between HuFNSC03 and HuFNSC04 in each cell type. __`r length(isoform_HuFNSC03_HuFNSC04_gene_shared)`__ genes are shared between two cell types.            

```{r individual, results='asis'}
library(xtable)
individual_summary <- xtable(individual_summary)
align(individual_summary) <- "lcccccc"
print(individual_summary, type = "html", include.rownames = T)

plot.new()
grid.draw(venn_HuFNSC01_HuFNSC02)
plot.new()
grid.draw(venn_HuFNSC03_HuFNSC04)
```

### Functional enrichment analysis 
#### HuFNSC01 vs HuFNSC02
  * Different regions on the Venn diagram have __no__ significantly enriched terms.     
  * Isoforms between HuFNSC01 and HuFNSC02 in different cell types show similar terms, related to signaling.     

```{r enrich_HuFNSC01_02, fig.width=10, fig.height = 4}
(isoform_brain01_brain02_enrich)
(isoform_cortex01_cortex02_enrich)
(isoform_GE01_GE02_enrich)
```

#### HuFNSC03 vs HuFNSC04
  
```{r enrich_HuFNSC03_04, fig.width=10, fig.height = 2}
(isoform_cortex03_cortex04_enrich)
(isoform_GE03_GE04_enrich)
```





