---
output: 
  html_document:
    keep_md: true
    toc: yes
---
Fetal Brain WGBS Analysis Summary - DMRs between Cortex and GE
========================================================

Gloria Li         
`r date()` 

```{r include = FALSE}
library(knitr)
library(ggplot2)
library(dplyr)
library(xtable)
library(VennDiagram)
load("~/快盘/FetalBrain/WGBS/WGBS.DMR.Rdata")
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

## DMR identification with methyl_diff

  * Identify DM CpGs     
    + methyl_diff one-sided p-value $\le$ 0.005  
    + delta fractional methylation $\ge$ 0.5  
    + fractional methylation of one sample $\ge$ 0.75   
  * Collapse DM CpGs into DMRs     
    + adjacent DM CpGs have the same DM status;    
    + distance between adjacent CpGs (size) $\le$ 300bp;   
    + No. of CpGs within each DMR $\ge$ 3.   

## Summary and sanity check  

  * On average, there are `r mean(DMR_WGBS_summary$Hypo.DMR)` Cortex UMRs, 179 intersect between two individuals, and `r mean(DMR_WGBS_summary$Hyper.DMR)` GE UMRs, 10 intersect. The intersect is significant. And there seems to be an asymmetry between Cortex UMRs and GE UMRs.    
  * Median DMR length is `r median(c(Cortex02_GE02_DMR$V7, Cortex04_GE04_DMR$V7))`, _comparable to breast_. It's similar in all chromosomes in Cortex UMRs, but fluctuate more in GE UMRs, probably due to  small No. of UMRs identified.   
  * Median No. of CpGs per DMR is `r median(c(Cortex02_GE02_DMR$V6, Cortex04_GE04_DMR$V6))`, _similar to breast_. chr11 and chr13 in GE UMRs have higher No. of CpGs per DMR.  

```{r WGBS_sanity, results='asis'}
DMR_WGBS_summary <- xtable(DMR_WGBS_summary)
align(DMR_WGBS_summary) <- "ccccc"
print(DMR_WGBS_summary, type = "html", include.rownames = F)

(valid_boxplot + ggtitle("Validate WGBS UMRs with MeDIP/MRE in HuFNSC02"))

(Cortex02_GE02_DMR_figures$length)
(Cortex02_GE02_DMR_figures$count)
plot.new()
grid.text("Venn diagram of Cortex UMRs", x = unit(0.5, "npc"), y = unit(0.9, "npc"))
grid.draw(venn_Cortex_UMR_WGBS)
plot.new()
grid.text("Venn diagram of GE UMRs", x = unit(0.5, "npc"), y = unit(0.9, "npc"))
grid.draw(venn_GE_UMR_WGBS)
```

## Asymmetry between Cortex UMRs and GE UMRs  

  * On average, there are , `r as.integer(mean(DMR_WGBS_summary$Hypo.DMR)/mean(DMR_WGBS_summary$Hyper.DMR))`-times more UMRs in Cortex than GE.  
  * The asymmetry appears to be global, in all chromosomes, and is reproduced in the two individuals.  
  * __Single CpG level__ differential methylation is __symmetric__, but the asymmetry on UMR level can be reproduced with __different cutoffs__. However, there are __no apparent differences in UMR length__ between Cortex and GE, suggesting that there are more __orphan GE UM CpGs__ that was not able to form UMRs than in Cortex. 

```{r WGBS_asymmetry}
(Cortex02_GE02_DMR_figures$freq)
(Cortex04_GE04_DMR_figures$freq)
(Cortex02_GE02_DMR_figures$pos + geom_point(aes(x = (as.numeric(chr) + 0.25*DM), y = pos, color = factor(DM, levels = c("1", "-1"))), position = position_jitter(width = 0.05), size = 1.5, alpha = 0.5))
(Cortex04_GE04_DMR_figures$pos + geom_point(aes(x = (as.numeric(chr) + 0.25*DM), y = pos, color = factor(DM, levels = c("1", "-1"))), position = position_jitter(width = 0.05), size = 1.5, alpha = 0.5))
(ggplot(DM_test_summary, aes(x = p, y = log2(Hyper.DM.CpGs/Hypo.DM.CpGs), color = Sample)) + scale_x_log10(breaks = c(0.05, 0.005, 0.0005)) + geom_point() + geom_line() + geom_hline(aes(yintercept = 0)) + facet_wrap(~ delta) + theme_bw() + xlab("p-value cutoff") + ylab("log2(Hyper/Hypo)"))
(ggplot(DMR_test_summary, aes(x = size, y = log2(Hyper.DMR/Hypo.DMR), color = Sample)) + geom_point() + geom_line() + geom_hline(aes(yintercept = 0)) + facet_grid(p ~ delta) + theme_bw() + xlab("Max distance between adjacent DM CpGs") + ylab("log2(Hyper/Hypo)"))
```

## GREAT analysis on Cortex UMRs and GE UMRs  

  * UMRs in both Cortex and GE in both individuals show enrichment in  __transcriptional regulation__ activities.  
  * In HuFNSC02, both Cortex UMRs are enriched in __brain regions development__, and GE UMRs are enriched in __neuron development__.   
  * In HuFNSC04, Cortex UMRs show __abnormal brain development__ in Mouse Phenotype, but are also enriched in __kidney-related processes__.   

```{r WGBS_GREAT1, fig.height=10, fig.width=8}
(GREAT_Cortex02.UMR)
```
```{r WGBS_GREAT2, fig.width=8}
(GREAT_GE02.UMR)
```
```{r WGBS_GREAT3, fig.height=12, fig.width=8}
(GREAT_Cortex04.UMR)
```
```{r WGBS_GREAT4, fig.height=4, fig.width=8}
(GREAT_GE04.UMR)
```

## UMR genomic break down  

  + On average, __`r round(summarise(filter(genomicBreak_WGBS_figure$data, Region == "Gene"), mean(CpG))*100, 2)`%__ of CpGs in UMRs overlap with genebody, and __`r round(summarise(filter(genomicBreak_WGBS_figure$data, Region == "Promoter"), mean(CpG))*100,2)`%__ of CpGs in UMRs overlap with promoters, not a significant enrichment __(`r round(summarise(filter(genomicBreak_WGBS_figure$data, Region == "Promoter"), mean(CpG))/(3727169/28217448), 2)`-fold)__.   

<!-- For the entire genome, 3727169 out of 28217448 CpGs overlap with TSS +/- 1500bp promoter regions -->

```{r WGBS_breakdown}
(genomicBreak_WGBS_figure + ggtitle("DM CpG breakdown between WGBS Cortex and GE"))
```

## UMRs intersecting with protein-coding genes and DE genes

  + On average, there are __`r as.integer(mean(DMR_WGBS_gene_summary$pc.Promoters))`__ DMRs associated with promoters of protein-coding genes, __`r round(mean(DMR_WGBS_gene_summary$pc.Promoters)/mean(DMR_WGBS_summary$Total.DMR)*100, 2)`%__ of all DMRs.         
  + On average, there are __`r as.integer(mean(DMR_WGBS_gene_summary$proximal.DE.Genes))`__ promoter DMRs associated with DE genes, __`r round(mean(DMR_WGBS_gene_summary$proximal.DE.Genes)/mean(DMR_WGBS_gene_summary$pc.Promoters)*100, 2)`%__ of all promoter DMRs. Among them, there are __`r round(mean(DMR_WGBS_gene_summary$same.direction)/mean(DMR_WGBS_gene_summary$unique.DE.Genes)*100, 2)`%__ unique DE genes change in the same direction as the UMRs.         
  + The intersect between two individuals are __significant__ when there are common genes. There are no intersect in pc genes with promoter GE UMRs.  
  + There are __no significant__ DAVID enrichment terms due to the small number of genes.  

```{r WGBS_pc, results='asis'}
DMR_WGBS_gene_summary <- xtable(DMR_WGBS_gene_summary)
align(DMR_WGBS_gene_summary) <- "cccccccc"
print(DMR_WGBS_gene_summary, type = "html", include.rownames = T)

plot.new()
grid.draw(venn_Cortex_UMR_pcGene_WGBS)
plot.new()
grid.draw(venn_GE_UMR_pcGene_WGBS)
plot.new()
grid.draw(venn_Cortex_UMR_pcPromoter_WGBS)
plot.new()
grid.draw(venn_Cortex_UMR_pcPromoter_DE_WGBS)
```

### DE genes with promoter Cortex UMRs  
#### HuFNSC02  
```{r WGBS_DE_Cortex02_UMR, results='asis'}
Cortex02_UMR_pcPromoter_DE_print <- xtable(select(Cortex02_UMR_pcPromoter_DE, name, description))
align(Cortex02_UMR_pcPromoter_DE_print) <- "rrr"
print(Cortex02_UMR_pcPromoter_DE_print, type = "html", include.rownames = F)
```

#### HuFNSC04  
```{r WGBS_DE_Cortex04_UMR, results='asis'}
Cortex04_UMR_pcPromoter_DE_print <- xtable(select(Cortex04_UMR_pcPromoter_DE, name, description))
align(Cortex04_UMR_pcPromoter_DE_print) <- "rrr"
print(Cortex04_UMR_pcPromoter_DE_print, type = "html", include.rownames = F)
```

### DE genes with promoter GE UMRs  
#### HuFNSC02  
```{r WGBS_DE_GE02_UMR, results='asis'}
GE02_UMR_pcPromoter_DE_print <- xtable(select(GE02_UMR_pcPromoter_DE, name, description))
align(GE02_UMR_pcPromoter_DE_print) <- "rrr"
print(GE02_UMR_pcPromoter_DE_print, type = "html", include.rownames = F)
```

#### HuFNSC04  
```{r WGBS_DE_GE04_UMR, results='asis'}
GE04_UMR_pcPromoter_DE_print <- xtable(select(GE04_UMR_pcPromoter_DE, name, description))
align(GE04_UMR_pcPromoter_DE_print) <- "rrr"
print(GE04_UMR_pcPromoter_DE_print, type = "html", include.rownames = F)
```



