File paths on xhost
===================

## Programs
* R: `/gsc/software/linux-x86_64-centos5/R-3.1.1/bin/R`  
* Java: `/gsc/software/linux-x86_64-centos5/java-1.7.0-u13/bin/java`
* Python: `/gsc/software/linux-x86_64-centos5/python-2.7.5/bin/python`    
* Matlab: `/gsc/software/linux-x86_64-centos5/matlab-2013a/bin/matlab`  
* Bedtools: `/projects/epigenomics/software/bedtools-2.23.0/bin/`
* Samtools: `/gsc/software/linux-x86_64-centos5/samtools-0.1.18/bin/samtools`    
* BCFtools: `/home/pubseq/BioSw/samtools/samtools-0.1.16/bcftools/bcftools`   
* Picard: `/gsc/software/linux-x86_64-centos5/picard-tools-1.92/`   
* IGV: `/gsc/software/linux-x86_64/IGV-2.3.47/igv.sh`       
* Homer: `/home/acarles/homer/bin/`
* ChromHMM: `/home/lli/bin/ChromHMM/ChromHMM.jar`      
* methyl_diff: `/home/mbilenky/methyl_diff-methyl_diff/methyl_diff`   
* Misha's java scripts: `/home/mbilenky/bin/Solexa_Java/`   
* Misha's matlab scripts: `/home/mbilenky/matlab/`  
* R libraries: `/home/lli/bin/R-3.1.1/`    
* R functions: `/home/lli/HirstLab/Pipeline/R/`
* Shell scripts: `/home/lli/HirstLab/Pipeline/shell/`   
* Python scripts: `/home/lli/HirstLab/Pipeline/python/`  
* UCSC scripts: `/home/lli/HirstLab/Pipeline/UCSC/`  

## Rescources
* Self processed hg19: `/home/lli/hg19/`
* Ensembl v65: `/projects/epigenomics/resources/Ensembl/hg19v65/`
* Genome hg19: `/home/pubseq/genomes/Homo_sapiens/hg19a/bwa_ind/genome/GRCh37-lite.fa`    
* Chromosome sizes: `/projects/epigenomics/resources/UCSC_chr/hg19.chrom.sizes`
* Encode TFBS: `/projects/mbilenky/REMC/breast/ENCODE/TFs/wgEncodeRegTfbsClusteredV3.bed.gz`    
* GWAS: `/projects/epigenomics/resources/UCSC_hg19/GWAS/` (bed `/home/lli/hg19/gwasCatalog_July2014.bed`)
* Mappability: `/projects/epigenomics/resources/UCSC_hg19/mappability/mappability_lt_1`
* CpG +/- 25bp bins: `/projects/epigenomics/users/mbilenky/CpG/hg19/CG_25_around_chr/`  
* Consortium 15 states model: `/home/lli/Consortium15states/model_15_coreMarks.txt`

## REMC project
* WGBS fractional calls: `/projects/edcc_prj2/bs-seq/`
* RNA-seq: `/projects/epigenomics/ep50/internal/jqc.1.7.6/`   
* Breast
    + Histone modification wig files: `/projects/mbilenky/REMC/breast/hg19/marks/wigs/`  
    + MeDIP wig files: `/projects/mbilenky/REMC/breast/hg19/MeDIP/wigs/`
    + Junction coverage: `/projects/mbilenky/REMC/breast/RNA-seq/junctions_new/`
    + UCSC browser hubs: `/gsc/www/bcgsc.ca/downloads/mb/BCHubs/`  
* Fetal Brain:
    + Histone modification wig files: `/home/lli/FetalBrain/HisMod/wigs/`
    + WGBS fractional calls and coverage symlink files: `/projects/epigenomics/users/lli/FetalBrain/WGBS/`
    + WGBS nonCpG methylation: `/projects/production_genomics/bs-seq/*/non_CpG*`        
    + MeDIP wig files: `/projects/mbilenky/REMC/brain/MeDIP/wigs/`  
    + MeDIP fractional calls: `/projects/epigenomics/users/lli/FetalBrain/MeDIP/`
    + DMRs from MeDIP: `/projects/epigenomics/users/lli/FetalBrain/MeDIP/DMR/`  
    + Junction coverage: `/home/lli/FetalBrain/RNAseq/junction/`
    + UCSC browser hubs: `/gsc/www/bcgsc.ca/downloads/mb/BrainHubs/`  
* H1:   
    + WGBS: `/projects/mbilenky/REMC/H1/hg19/WGBS/` 
    + Histone modifications: `/projects/mbilenky/REMC/H1/UCSF_UBC/wigs_175/`   
    + RNA-seq: `/projects/epigenomics/ep50/external/jqc.1.7.6/`   
* ESC:
    + RNA-seq: `/projects/epigenomics/ep50/external/jqc.1.7.6/`

## CEMT project
* Project folder: `/projects/edcc_new/reference_epigenomes/`     
* WGBS fractional methylation calls: `/projects/edcc_prj2/bs-seq/<libraryID>/*5mC.CpG`    
* WGS SNPs: `/projects/edcc_new/reference_epigenomes/<libraryID>/bams/WGS/*COSM*vcf`      
