Analysis pipeline
====================
* [Path.md](./Path.md): file paths on xhost.    
* [DEfine.md](./DEfine.md): instructions and sample code for using DEfine on genes or exons.  
* [isoform.md](./isoform.md): instructions for identifying isoform genes and cassette exons.  
* [junction.md](./junction.md): instructions for validating isoforms with junction reads.  
* [epiProfile.md](./epiProfile.md): instructions for calculating and plotting epigenetic profiles for isoform and intron retention.       
* [ChromHMM.md](./ChromHMM.md): instructions on running ChromHMM. _unfinished_.  
* [MeDIP.DMR.py](./MeDIP.DMR.py): collapse single DM CpGs to DMRs.
* [DMR.figures.R](./DMR.figures.R): `DMR_figures` R function for DMR analysis and visualization.   
* [enrich.R](./enrich.R): `enrich` R function to plot enriched terms with `ggplot2`.   
* [isoform.R](./isoform.R): `isoform` R function for identifying isoform genes and cassette exons.
* [junction.R](./junction.R): `junction` R function for validating isoform genes and cassette exons with junction reads.   
* [epiProfile.R](./epiProfile.R): `epiProfile` R function for calculating and plotting epigenetic profiles for isoform and intron retention.       
* [DEfine.m](./DEfine.m): copy of Misha's DEfine function.   
* [myfnorm.m](./myfnorm.m): copy of Misha's myfnorm function, required for DEfine.
* [UCSC](./UCSC): script from UCSC genome browser and sample code for building track hub.

