---
title: "VitC - ChIPseq"
author: "Gloria Li"
date: "Sept 6, 2017"
output: 
  html_document:
    keep_md: true
    toc: true
    toc_float: true
    toc_depth: 4
---

Updated: `r date()`

```{r setup, include = FALSE}
library(ggplot2)
library(plyr)
library(VennDiagram)
library(gridExtra)
library(gplots)
library(dendextend)
library(reshape)
library(wq)
library(dplyr)
library(RCircos)
library(knitr)
load("/projects/epigenomics3/epigenomics3_results/users/lli/NHA/ChIPseq/ChIPseq.Rdata")
knitr::opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE, results = FALSE, fig.height = 6, fig.width = 6) 
knitr::opts_knit$set(root.dir = "/projects/epigenomics3/epigenomics3_results/users/lli/NHA/ChIPseq/")
```

## QC  
* Sequencing depth are even across samples, and sufficient (50M for narrow marks, 100M for broad marks).     
* All QC metric looks good.       
* All results suggest that NHAR VitC H3K9me3 sample is actually a H3K4me3 sample, probably due to using the wrong antibody in IP: remove H3K9me3 from further analysis.          

```{r QC}
(QC_summary_reads_figure)
(QC_summary_percent_figure)
(QC_summary_domain_figure)
```

## Enrich regions
* From [UCSC tracks](http://www.bcgsc.ca/downloads/mb/VitC_glioma/HistoneHub/hub.txt), FindER results make more sense than MACS2.      

```{r ER_summary}
(ER_summary_FindER_figure)
```

## Unique enrich regions
* Unique enrich regions were identified by non-overlapping regions in pairwise comparisons.    

```{r ER_unique_summary}
(ER_unique_summary_figure)
```

### Homer for unique H3K27ac 
* Results correlate those in CEMT glioma.    

```{r homer_K27ac1, fig.height=4}
(homer_unique_enhancer_NHARcontrol_NHARvitc_figure)
```
```{r homer_K27ac2, fig.height=2}
(homer_unique_enhancer_NHARvitc_NHARcontrol_figure)
```
```{r homer_K27ac3, fig.height=4}
(homer_unique_enhancer_NHARcontrol_NHAcontrol_figure)
```
```{r homer_K27ac4, fig.height=2}
(homer_unique_enhancer_NHAcontrol_NHARcontrol_figure)
```

