---
output:
  html_document:
    keep_md: true
    toc: yes
---
Fetal Brain isoform analysis - junctions
========================================================

Gloria Li         
`r date()` 

<!-- re-knit after modify junction.std.R script -->

```{r include = FALSE}
library(knitr)
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

```{r data}
setwd("~/快盘/FetalBrain/RNAseq/junction")
load("FetalBrain_isoform_valid.Rdata")
library(ggplot2)
```

## Validate previously identified isoforms with junction RPKM
  * Previous isoform identification with DE exons
    * DE exons by DEfine FDR = 0.01
    * Exon RPKM $\ge$ 10% gene RPKM in one sample & $\le$ 1% in the other
    * Gene RPKM of both samples > 0.005 
    * Exclude DE genes by DEfine FDR = 0.01

  * Validation: For each isoform exon in the previous pairwise comparison
    * Find junctions associated with this exon with enough coverage, i.e. sum of junction coverage of two samples $\ge$ 1
    * Identify junctions that RPKM change in the same direction as the exon
    * Junction RPKM > 0.1 in one sample and < 0.1 in the other      
  
## Results: 
### No. of exons for DE genes / isoform genes    
  * DE genes have roughly the same No. of exons as all expressed genes.             
  * Identified isoforms have slightly more No. of exons than DE genes and all expressed genes.  
  * Compared to DE genes, the distribution in No. of exons for isoforms are __much similar__ between different individuals, _not observed in breast libraries_.    
  
```{r Nexon, fig.height = 8, fig.width = 8}
plot(c(0, 50), c(0, 0.08), type = "n", main = "No. of exons in DE genes and isoforms", xlab = "No. of exons", ylab = "density")
lines(density(na.omit(Nexon[rownames(Nexon) %in% as.character(geneRPKM[(geneRPKM$cortex01 + geneRPKM$cortex02 + geneRPKM$cortex03 + geneRPKM$cortex04 + geneRPKM$ge01 + geneRPKM$ge02 + geneRPKM$ge03 + geneRPKM$ge04) > 0.08, "id"]), "Nexon"])), col = 1, lty = 1, lwd = 5)
lines(density(na.omit(cortex01_GE01_DE$Nexon)), col = "red", lty = 1, lwd = 3)
lines(density(na.omit(cortex02_GE02_DE$Nexon)), col = "blue", lty = 1, lwd = 3)
lines(density(na.omit(cortex03_GE03_DE$Nexon)), col = "green", lty = 1, lwd = 3)
lines(density(na.omit(cortex04_GE04_DE$Nexon)), col = "purple", lty = 1, lwd = 3)
lines(density(na.omit(cortex01_GE01_isoform_gene$Nexon)), col = "red", lty = 3, lwd = 3)
lines(density(na.omit(cortex02_GE02_isoform_gene$Nexon)), col = "blue", lty = 3, lwd = 3)
lines(density(na.omit(cortex03_GE03_isoform_gene$Nexon)), col = "green", lty = 3, lwd = 3)
lines(density(na.omit(cortex04_GE04_isoform_gene$Nexon)), col = "purple", lty = 3, lwd = 3)
legend("topleft", c("all expressed genes", "DE genes", "isoforms"), col = 1, lty = c(1, 1, 3), lwd = 5, cex = 0.8)
legend("topright", c("all expressed genes", "cortex01 vs GE01", "cortex02 vs GE02", "cortex03 vs GE03", "cortex04 vs GE04"), col = c("black", "red", "blue", "green", "purple"), lty = 1, lwd = 5, cex = 0.8)
```

### Position of isoform exons on the gene   
  * In general, there are more alternative spliced exons at the __two ends__ of genes, _similar to observed in breast libraries_.         
  
```{r exon_pos, fig.height = 8, fig.width = 8}
plot(c(0, 100), c(0, 0.015), type = "n", main = "Distribution of isoform exons along genes", xlab = "gene position", ylab = "density")
lines(density(na.omit(cortex01_GE01_isoform_exon$exon_pos)), col = "red", lty = 1, lwd = 3)
lines(density(na.omit(cortex02_GE02_isoform_exon$exon_pos)), col = "blue", lty = 1, lwd = 3)
lines(density(na.omit(cortex03_GE03_isoform_exon$exon_pos)), col = "green", lty = 1, lwd = 3)
lines(density(na.omit(cortex04_GE04_isoform_exon$exon_pos)), col = "purple", lty = 1, lwd = 3)
legend("bottomright", c("cortex01 vs GE01", "cortex02 vs GE02", "cortex03 vs GE03", "cortex04 vs GE04"), col = c("red", "blue", "green", "purple"), lty = 1, lwd = 5)
```

### Junction validation     
  + For cortex vs GE, on average, __`r round(mean(cortex_GE_valid_summary[, "genes with junction coverage"]/cortex_GE_valid_summary[, "isoform genes"])*100, 1)`%__ isoform genes have enough junction coverage. Among them, __`r round(mean(cortex_GE_valid_summary[, "genes with junction support"]/cortex_GE_valid_summary[, "genes with junction coverage"])*100, 1)`%__ have support from junction reads.    
  + For comparisons between individuals, on average, __`r round(mean(individual_valid_summary[, "genes with junction coverage"]/individual_valid_summary[, "isoform genes"])*100, 1)`%__ isoform genes have enough junction coverage. Among them, __`r round(mean(individual_valid_summary[, "genes with junction support"]/individual_valid_summary[, "genes with junction coverage"])*100, 1)`%__ have support from junction reads.     
  + Between strand specific and non-strand specific libraries, the percentage of isoforms with enough junction coverage are __similar__, however, strand specific libraries have __higher__ percentage of having junction support(> 98% compared to ~ 80%).   
  + From the Venn diagrams, similar to observed in breast libraries, the overlapping isoforms between different comparisons have __much lower__ ratio of being validated compared to comparison specific isoforms.    

```{r validate, results='asis', fig.height = 5, fig.width = 5}
library(xtable)
cortex_GE_valid_summary <- xtable(cortex_GE_valid_summary)
align(cortex_GE_valid_summary) <- "lcccccc"
print(cortex_GE_valid_summary, type = "html", include.rownames = T)
individual_valid_summary <- xtable(individual_valid_summary)
align(individual_valid_summary) <- "lcccccc"
print(individual_valid_summary, type = "html", include.rownames = T)

library(VennDiagram)
# cortex vs GE
plot.new()
grid.draw(venn_cortex_GE)
plot.new()
grid.draw(venn_cortex_GE_valid)

# HuFNSC01 vs HuFNSC02
plot.new()
grid.draw(venn_HuFNSC01_HuFNSC02)
plot.new()
grid.draw(venn_HuFNSC01_HuFNSC02_valid)

# HuFNSC03 vs HuFNSC04
plot.new()
grid.draw(venn_HuFNSC03_HuFNSC04)
plot.new()
grid.draw(venn_HuFNSC03_HuFNSC04_valid)
```

### Venn Diagram with average expression level, average No. of exons and average exon length   
  * Isoforms have __much lower__ expression level than all expressed genes.          
  * On average, common isoforms between different comparisons have __lower expression level__ than comparison-specific isoforms.                 
  * In general, compared to all isoforms identified, validated isoforms have __lower__ expression levels, _not observed in breast libraries_.     
  * Average No. of exons are very __similar__ in different sections of the Venn diagram, between all, validated isoforms and all expressed genes.        
  * Average length of isoform exons are __shorter__ than all expressed genes. Validated isoform exons are also __shorter__ than all isoforms in general, _not observed in breast libraries_.         
  
```{r venn, fig.height = 5, fig.width = 5}
# No. of isoforms
(bubble_all_N)
venn_all <- venn.diagram(isoform_all, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "No. of isoform genes", main.cex = 1.5)
grid.draw(venn_all)
(bubble_valid_N)
venn_valid <- venn.diagram(isoform_valid, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "No. of validated isoform genes", main.cex = 1.5)
grid.draw(venn_valid)

# average expression level
(bubble_all_rpkm)
venn_all <- venn.diagram(isoform_all, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "Average gene RPKM of isoforms", main.cex = 1.5)
grid.draw(venn_all)
(bubble_valid_rpkm)
venn_valid <- venn.diagram(isoform_valid, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "Average gene RPKM of validated isoforms", main.cex = 1.5)
grid.draw(venn_valid)

# No. of exons
(bubble_all_Nexon)
venn_all <- venn.diagram(isoform_all, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "Average No. of exons of isoforms", main.cex = 1.5)
grid.draw(venn_all)
(bubble_valid_Nexon)
venn_valid <- venn.diagram(isoform_valid, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "Average No. of exons of validated isoforms", main.cex = 1.5)
grid.draw(venn_valid)

# average exon length 
(bubble_all_exon_length)
venn_all <- venn.diagram(isoform_all, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "Average length of isoform exons", main.cex = 1.5)
grid.draw(venn_all)
(bubble_valid_exon_length)
venn_valid <- venn.diagram(isoform_valid, filename = NULL, fill = rep("white", 3), cex = 0, alpha = 0, col = c("red", "blue", "green"), main = "Average length of validated isoform exons", main.cex = 1.5)
grid.draw(venn_valid)
```

