Fetal Brain isoform analysis - junctions
========================================================

Gloria Li         
`r date()` 

<!-- re-knit after modify junction_valid.R or junction.R script -->

```{r include = FALSE}
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

```{r data}
setwd("~/快盘/FetalBrain/RNAseq/junction")
load("junction_valid.Rdata")
library(ggplot2)
```

## Validate previously identified isoforms with junction RPKM
  * Previous isoform identification with DE exons
    * DE exons by DEfine FDR = 0.01
    * Exon RPKM $\ge$ 10% gene RPKM in one sample & $\le$ 1% in the other
    * Gene RPKM of both samples > 0.005 
    * Exclude DE genes by DEfine FDR = 0.01

  * Validation: For each isoform exon in the previous pairwise comparison
    * Find junctions associated with this exon with enough coverage, i.e. sum of junction coverage of two samples $\ge$ 1
    * Identify junctions that RPKM change in the same direction as the exon
    * Junction RPKM > 0.1 in one sample and < 0.1 in the other      
  
## Results: 
  + For cortex vs GE, on average, __`r round(mean(junction_valid_cortexge[, "No.genes.with.junction.cov"]/junction_valid_cortexge[, "No.isoform.genes"])*100, 1)`%__ isoform genes have enough junction coverage. Among them, __`r round(mean(junction_valid_cortexge[, "No.genes.with.junction.support"]/junction_valid_cortexge[, "No.genes.with.junction.cov"])*100, 1)`%__ have support from junction reads.    
  + For comparisons between individuals, on average, __`r round(mean(junction_valid_individual[, "No.genes.with.junction.cov"]/junction_valid_individual[, "No.isoform.genes"])*100, 1)`%__ isoform genes have enough junction coverage. Among them, __`r round(mean(junction_valid_individual[, "No.genes.with.junction.support"]/junction_valid_individual[, "No.genes.with.junction.cov"])*100, 1)`%__ have support from junction reads.     
  + Between strand specific and non-strand specific libraries, the percentage of isoforms with enough junction coverage are __similar__, however, strand specific libraries have __higher__ percentage of having junction support(> 98% compared to ~ 80%).   
  + From the Venn diagrams, similar to observed in breast libraries, the overlapping isoforms between different comparisons have __much lower__ ratio of being validated compared to comparison specific isoforms.    

```{r validate, results='asis'}
library(xtable)
junction_valid_cortexge <- xtable(junction_valid_cortexge)
align(junction_valid_cortexge) <- "lcccccc"
print(junction_valid_cortexge, type = "html", include.rownames = T)
junction_valid_individual <- xtable(junction_valid_individual)
align(junction_valid_individual) <- "lcccccc"
print(junction_valid_individual, type = "html", include.rownames = T)

library(VennDiagram)
# cortex vs GE
isoform_cortexge_all <- list(HuFNSC01 = HuFNSC01_isoform_all_gene$id, HuFNSC02 = HuFNSC02_isoform_all_gene$id, HuFNSC03 = HuFNSC03_isoform_all_gene$id, HuFNSC04 = HuFNSC04_isoform_all_gene$id)
plot.new()
venn_cortexge_all <- venn.diagram(isoform_cortexge_all, filename = NULL, fill = c("red", "blue", "green", "yellow"), main = "Venn diagram of all cortex vs GE isoforms")
grid.draw(venn_cortexge_all)
isoform_cortexge_valid <- list(HuFNSC01 = as.character(HuFNSC01_isoform_valid_gene$gene), HuFNSC02 = as.character(HuFNSC02_isoform_valid_gene$gene), HuFNSC03 = as.character(HuFNSC03_isoform_valid_gene$gene), HuFNSC04 = as.character(HuFNSC04_isoform_valid_gene$gene))
plot.new()
venn_cortexge_valid <- venn.diagram(isoform_cortexge_valid, filename = NULL, fill = c("red", "blue", "green", "yellow"), main = "Venn diagram of validated cortex01 vs GE01 isoforms")
grid.draw(venn_cortexge_valid)

# HuFNSC01 vs HuFNSC02
isoform_01_02_all <- list(Brain = unique(brain01_02_isoform_all$id), Cortex = unique(cortex01_02_isoform_all$id), GE = unique(ge01_02_isoform_all$id))
plot.new()
venn_01_02_all <- venn.diagram(isoform_01_02_all, filename = NULL, fill = c("red", "blue", "green"), main = "Venn diagram of all HuFNSC01 vs HuFNSC02 isoforms")
grid.draw(venn_01_02_all)
isoform_01_02_valid <- list(Brain = as.character(brain01_02_isoform_valid_gene$gene), Cortex = as.character(cortex01_02_isoform_valid_gene$gene), GE = as.character(ge01_02_isoform_valid_gene$gene))
plot.new()
venn_01_02_valid <- venn.diagram(isoform_01_02_valid, filename = NULL, fill = c("red", "blue", "green"), main = "Venn diagram of validated HuFNSC01 vs HuFNSC02 isoforms")
grid.draw(venn_01_02_valid)
```

### No. of exons for DE genes / isoform genes    
  * DE genes have roughly the same No. of exons as all expressed genes.             
  * Identified isoforms have slightly more No. of exons than DE genes and all expressed genes.  
  * Compared to DE genes, the distribution in No. of exons for isoforms are __much similar__ between different individuals.    
  
```{r Nexon}
plot(c(0, 50), c(0, 0.08), type = "n", main = "No. of exons in DE genes and isoforms", xlab = "No. of exons", ylab = "density")
lines(density(na.omit(Nexon[Nexon$id %in% geneRPKM[(geneRPKM$cortex01 + geneRPKM$cortex02 + geneRPKM$cortex03 + geneRPKM$cortex04 + geneRPKM$ge01 + geneRPKM$ge02 + geneRPKM$ge03 + geneRPKM$ge04) > 0.05,]$id,]$Nexon)), col = 1, lty = 1, lwd = 5)
lines(density(na.omit(HuFNSC01_DE$Nexon)), col = 2, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC02_DE$Nexon)), col = 3, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC03_DE$Nexon)), col = 4, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC04_DE$Nexon)), col = 5, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC01_isoform_all_gene$Nexon)), col = 2, lty = 3, lwd = 3)
lines(density(na.omit(HuFNSC02_isoform_all_gene$Nexon)), col = 3, lty = 3, lwd = 3)
lines(density(na.omit(HuFNSC03_isoform_all_gene$Nexon)), col = 4, lty = 3, lwd = 3)
lines(density(na.omit(HuFNSC04_isoform_all_gene$Nexon)), col = 5, lty = 3, lwd = 3)
legend("topleft", c("all expressed genes", "DE genes", "isoforms"), col = 1, lty = c(1, 1, 3), lwd = 5, cex = 0.8)
legend("topright", c("all expressed genes", "cortex01 vs GE01", "cortex02 vs GE02", "cortex03 vs GE03", "cortex04 vs GE04"), col = 1:5, lty = 1, lwd = 5, cex = 0.8)
```

### Position of isoform exons on the gene   
  * In general, there are more alternative spliced exons at the __two ends__ of genes, similar to observed in breast libraries.         
  
```{r exon_pos}
plot(c(0, 100), c(0, 0.015), type = "n", main = "Distribution of isoform exons along genes", xlab = "gene position", ylab = "density")
lines(density(na.omit(HuFNSC01_isoform_all$exon_pos)), col = 2, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC02_isoform_all$exon_pos)), col = 3, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC03_isoform_all$exon_pos)), col = 4, lty = 1, lwd = 3)
lines(density(na.omit(HuFNSC04_isoform_all$exon_pos)), col = 5, lty = 1, lwd = 3)
legend("bottomright", c("cortex01 vs GE01", "cortex02 vs GE02", "cortex03 vs GE03", "cortex04 vs GE04"), col = 2:5, lty = 1, lwd = 5)
```


