---
title: "VitC - PBAL"
author: "Gloria Li"
date: "Feb 18, 2018"
output: 
  html_document:
    keep_md: true
    toc: true
    toc_float: true
    toc_depth: 4
---

Updated: `r date()`

```{r setup, include = FALSE}
library(ggplot2)
library(plyr)
library(VennDiagram)
library(gridExtra)
library(gplots)
library(dendextend)
library(reshape)
library(wq)
library(dplyr)
library(RCircos)
library(knitr)
load("/projects/epigenomics3/epigenomics3_results/users/lli/NHA/PBAL/PBAL.Rdata")
knitr::opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE, results = FALSE, fig.height = 6, fig.width = 6) 
knitr::opts_knit$set(root.dir = "/projects/epigenomics3/epigenomics3_results/users/lli/NHA/PBAL/")
```

## QC
* Duplicate rate are higher than expected.      
* Coverage at CpG sites are lower than other libraries.     

```{r QC, fig.height=8}
(QC_summary_reads_figure)
(QC_summary_percent_figure)
(qc_5mC_coverage_figure)
```

## Clustering
* Very high correlation between samples within the same cell lines.    

```{r cluster}
op <- par(mar = c(5, 4, 4, 8))
plot(pearson_genome_5mC_dend, main = "genome-wide pearson", horiz = TRUE)
plot(pearson_CGI_5mC_dend, main = "CGI pearson", horiz = TRUE)
par(op)
```

## 5mC distribution
* No CIMP phenotype in NHAR samples, even though IDH mutant were expressed.   

```{r distribution}
(qc_5mC_quantile_figure)
```

### Why no CIMP in NHAR? 
* Hypothesis 1: IDH1 mutation unsuccefully induced -> NO     

```{r IDH_mut}
mut_rate <- read.delim("/projects/epigenomics3/epigenomics3_results/users/lli/NHA/IDH.mutant.rate.txt") %>% filter(Assay == "RNAseq")
kable(mut_rate, format = "html", align = "l", row.names = F)
```

* Hypothesis 2: TET mutation resulting in insensitive to 2HG -> Not likely       
	+ The only non-neutral variance in TETs identified in NHA lines is TET1 K1019E. It is outside TET1's 2HG, metal or DNA-binding domains, and is never annotated in cancer before.        
	+ 5hmC changes still occured as expected, suggessting TET is functioning.      

* Hypothesis 3: cells blocked at hmC, and can not transition to a CpG state -> not enough evidence        
	+ There are more regions lose 5hmC than gain from wt to mut, with no change in PBAL.      
	+ However, no evidence of 5hmC was found around hyper CpGs (+/- 100bp) identified in the Nature Genetics paper.    

```{r DMR_DhMR_summary}
DMR_DhMR_summary <- read.delim("../wt_mut_vitc/methylation/DMR.DhMR.summary.stats", as.is = T) %>% filter(Comparison == "wt-mut", PBAL == "ST")
kable(DMR_DhMR_summary, format = "html", align = "r", row.names = F)
```

![](./PBAL_files/figure-html/DMR.wt-mut.hMeDIP.png)      

* Hypotheis 4: CNVs (from PBAL -> depth too low)     
	+ MGG119     
![](./PBAL_files/figure-html/MGG_control.6_lanes_dupsFlagged.bam_ratio.txt.png)   

	+ NHA       
![](./PBAL_files/figure-html/NHA_control.6_lanes_dupsFlagged.bam_ratio.txt.png)   

	+ NHAR      
![](./PBAL_files/figure-html/NHAR_control.6_lanes_dupsFlagged.bam_ratio.txt.png)   

## DMRs
### Methods
* DM CpGs      
	+ Methyl_diff p-value < 0.1       
	+ delta 5mC > 0.25       
	+ hyper 5mC > 0.5     
* DMR dynamic growth     
	+ distance to adjacent DM CpG < 500bp      
	+ No. of CpGs > 3       

### Summary

```{r summary1}
DMR_summary <- read.delim("DMR/DMR.summary.stats") %>% arrange(Sample1, Sample2)
kable(DMR_summary, format = "html", align = "l", row.names = F)
```

* Remove differentially 5hmC regions from DMRs    

```{r summary2}
DMR_summary2 <- read.delim("DMR/DMR-DhMR.summary.stats") %>% arrange(Sample1, Sample2)
kable(DMR_summary2, format = "html", align = "l", row.names = F)
```

### Enrichment in genomic regions 

```{r genomeBreak}
(genomic_breakdown_figure)
```


