---
output: 
  html_document:
    keep_md: true
    toc: yes
---
Fetal Brain RNA-seq - DE genes 
========================================================

Gloria Li         
Updated: `r date()` 

```{r setup, include = FALSE}
library(knitr)
library(ggplot2)
library(xtable)
library(VennDiagram)
# source("~/HirstLab/Pipeline/enrich.R")
load("~/快盘/hg19/hg19v65_genes.Rdata")
load("~/快盘/FetalBrain/RNAseq/DEfine/gene/FetalBrain_DEgenes.Rdata")
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

```{r DEfine, eval=F}
col <- c("Gene id", "rpkm1", "rpkm2", "p-value", "Multiple testing corrected p-value")
setwd("~/快盘/FetalBrain/RNAseq/DEfine/gene/cortexge/")
cortex01_GE01UP<-read.table("UP.Cortex-HuFNSC01_GE-HuFNSC01.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex01_GE01UP$DE <- "UP"
cortex01_GE01DN<-read.table("DN.Cortex-HuFNSC01_GE-HuFNSC01.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex01_GE01DN$DE <- "DN"
cortex02_GE02UP<-read.table("UP.Cortex-HuFNSC02_GE-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex02_GE02UP$DE <- "UP"
cortex02_GE02DN<-read.table("DN.Cortex-HuFNSC02_GE-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex02_GE02DN$DE <- "DN"
cortex03_GE03UP<-read.table("UP.Cortex-HuFNSC03_GE-HuFNSC03.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex03_GE03UP$DE <- "UP"
cortex03_GE03DN<-read.table("DN.Cortex-HuFNSC03_GE-HuFNSC03.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex03_GE03DN$DE <- "DN"
cortex04_GE04UP<-read.table("UP.Cortex-HuFNSC04_GE-HuFNSC04.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex04_GE04UP$DE <- "UP"
cortex04_GE04DN<-read.table("DN.Cortex-HuFNSC04_GE-HuFNSC04.FDR_0.01.rmin_0.005.Nmin_25", as.is = T)
cortex04_GE04DN$DE <- "DN"
cortex01_GE01DE <- rbind(cortex01_GE01UP,cortex01_GE01DN)
rownames(cortex01_GE01DE) <- cortex01_GE01DE$V1
cortex02_GE02DE <- rbind(cortex02_GE02UP,cortex02_GE02DN)
rownames(cortex02_GE02DE) <- cortex02_GE02DE$V1
cortex03_GE03DE <- rbind(cortex03_GE03UP,cortex03_GE03DN)
rownames(cortex03_GE03DE) <- cortex03_GE03DE$V1
cortex04_GE04DE <- rbind(cortex04_GE04UP,cortex04_GE04DN)
rownames(cortex04_GE04DE) <- cortex04_GE04DE$V1
cortex_GE_DE_summary <- data.frame(UP = c(nrow(cortex01_GE01UP), nrow(cortex02_GE02UP), nrow(cortex03_GE03UP), nrow(cortex04_GE04UP)), DN = c(nrow(cortex01_GE01DN), nrow(cortex02_GE02DN), nrow(cortex03_GE03DN), nrow(cortex04_GE04DN)), DE = c(nrow(cortex01_GE01DE), nrow(cortex02_GE02DE), nrow(cortex03_GE03DE), nrow(cortex04_GE04DE)))
rownames(cortex_GE_DE_summary) <- c("HuFNSC01", "HuFNSC02", "HuFNSC03", "HuFNSC04")
cortex_GE_UP_duplicated <- c(cortex01_GE01UP$V1, cortex02_GE02UP$V1, cortex03_GE03UP$V1, cortex04_GE04UP$V1)
cortex_GE_UP_duplicated <- ensembl[unique(cortex_GE_UP_duplicated[duplicated(cortex_GE_UP_duplicated)]), ]
# write.table(cortex_GE_UP_duplicated, file = "~/快盘/FetalBrain/RNAseq/DEfine/gene/cortex_GE_UP_duplicated.txt", sep = "\t", quote = F, row.names = F, col.names = F)
cortex_GE_DN_duplicated <- c(cortex01_GE01DN$V1, cortex02_GE02DN$V1, cortex03_GE03DN$V1, cortex04_GE04DN$V1)
cortex_GE_DN_duplicated <- ensembl[unique(cortex_GE_DN_duplicated[duplicated(cortex_GE_DN_duplicated)]), ]
# write.table(cortex_GE_DN_duplicated, file = "~/快盘/FetalBrain/RNAseq/DEfine/gene/cortex_GE_DN_duplicated.txt", sep = "\t", quote = F, row.names = F, col.names = F)

setwd("~/快盘/FetalBrain/RNAseq/DEfine/gene/individual/")
brain01_brain02UP <- read.table("UP.Brain-HuFNSC01_Brain-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25")
brain01_brain02UP$DE <- "UP"
brain01_brain02DN <- read.table("DN.Brain-HuFNSC01_Brain-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25")
brain01_brain02DN$DE <- "DN"
cortex01_cortex02UP <- read.table("UP.Cortex-HuFNSC01_Cortex-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25")
cortex01_cortex02UP$DE <- "UP"
cortex01_cortex02DN <- read.table("DN.Cortex-HuFNSC01_Cortex-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25")
cortex01_cortex02DN$DE <- "DN"
cortex03_cortex04UP <- read.table("UP.Cortex-HuFNSC03_Cortex-HuFNSC04.FDR_0.01.rmin_0.005.Nmin_25")
cortex03_cortex04UP$DE <- "UP"
cortex03_cortex04DN <- read.table("DN.Cortex-HuFNSC03_Cortex-HuFNSC04.FDR_0.01.rmin_0.005.Nmin_25")
cortex03_cortex04DN$DE <- "DN"
GE01_GE02UP <- read.table("UP.GE-HuFNSC01_GE-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25")
GE01_GE02UP$DE <- "UP"
GE01_GE02DN <- read.table("DN.GE-HuFNSC01_GE-HuFNSC02.FDR_0.01.rmin_0.005.Nmin_25")
GE01_GE02DN$DE <- "DN"
GE03_GE04UP <- read.table("UP.GE-HuFNSC03_GE-HuFNSC04.FDR_0.01.rmin_0.005.Nmin_25")
GE03_GE04UP$DE <- "UP"
GE03_GE04DN <- read.table("DN.GE-HuFNSC03_GE-HuFNSC04.FDR_0.01.rmin_0.005.Nmin_25")
GE03_GE04DN$DE <- "DN"
brain01_brain02DE <- rbind(brain01_brain02UP,brain01_brain02DN)
rownames(brain01_brain02DE) <- brain01_brain02DE$V1
cortex01_cortex02DE <- rbind(cortex01_cortex02UP,cortex01_cortex02DN)
rownames(cortex01_cortex02DE) <- cortex01_cortex02DE$V1
cortex03_cortex04DE <- rbind(cortex03_cortex04UP,cortex03_cortex04DN)
rownames(cortex03_cortex04DE) <- cortex03_cortex04DE$V1
GE01_GE02DE <- rbind(GE01_GE02UP,GE01_GE02DN)
rownames(GE01_GE02DE) <- GE01_GE02DE$V1
GE03_GE04DE <- rbind(GE03_GE04UP,GE03_GE04DN)
rownames(GE03_GE04DE) <- GE03_GE04DE$V1
individual_DE_summary <- data.frame(UP = c(nrow(brain01_brain02UP), nrow(cortex01_cortex02UP), nrow(GE01_GE02UP), nrow(cortex03_cortex04UP), nrow(GE03_GE04UP)), DN = c(nrow(brain01_brain02DN), nrow(cortex01_cortex02DN), nrow(GE01_GE02DN), nrow(cortex03_cortex04DN), nrow(GE03_GE04DN)), DE = c(nrow(brain01_brain02DE), nrow(cortex01_cortex02DE), nrow(GE01_GE02DE), nrow(cortex03_cortex04DE), nrow(GE03_GE04DE)))
rownames(individual_DE_summary) <- c("brain01_brain02", "cortex01_cortex02", "GE01_GE02", "cortex03_GE03", "cortex04_GE04")
# write.table(brain01_brain02DE, file = "~/快盘/FetalBrain/RNAseq/DEfine/gene/brain01_brain02DE.txt", sep = "\t", quote = F, row.names = F, col.names = F)
# write.table(cortex01_cortex02DE, file = "~/快盘/FetalBrain/RNAseq/DEfine/gene/cortex01_cortex02DE.txt", sep = "\t", quote = F, row.names = F, col.names = F)
# write.table(GE01_GE02DE, file = "~/快盘/FetalBrain/RNAseq/DEfine/gene/GE01_GE02DE.txt", sep = "\t", quote = F, row.names = F, col.names = F)
```

## Differentially expressed genes
### DEfine

  * FDR = 0.01    
  * Minimum sum of RPKM (rmin) = 0.005    
  * Minimum sum of coverage (Nmin) = 25    
  
### Between cortex and GE neurospheres

  * On average, there are __`r mean(cortex_GE_DE_summary$DE)`__ genes differentially expressed between cortex and GE, among them, __`r mean(cortex_GE_DE_summary$UP)`__ are upregulated in cortex, and __`r mean(cortex_GE_DE_summary$DN)`__ are downregulated.    
  * __`r nrow(cortex_GE_UP_duplicated)`__ Cortex up-regulated genes, and __`r nrow(cortex_GE_DN_duplicated)`__ GE up-regulated genes are shared by at least two individuals.    
  * DAVID enrichment analysis show significant enrichment in __neuronal development__ and __cell migration__ terms, GE up-regulated genes are enriched in __EGF-related__ protein domains as well. 

```{r cortex_GE, results='asis', fig.height = 6, fig.width = 6}
setwd("~/快盘/FetalBrain/RNAseq/DEfine/gene/")
cortex_GE_DE_summary <- xtable(cortex_GE_DE_summary)
align(cortex_GE_DE_summary) <- "lccc"
print(cortex_GE_DE_summary, type = "html", include.rownames = T)

# UP_cortex_GE <- list(HuFNSC01 = cortex01_GE01UP$V1, HuFNSC02 = cortex02_GE02UP$V1, HuFNSC03 = cortex03_GE03UP$V1, HuFNSC04 = cortex04_GE04UP$V1)
# venn_UP_cortex_GE <- venn.diagram(UP_cortex_GE, filename = NULL, fill = c("red", "blue", "green", "yellow"), main = "Venn diagram of Cortex up-regulated genes")
plot.new()
grid.draw(venn_UP_cortex_GE)
# DN_cortex_GE <- list(HuFNSC01 = cortex01_GE01DN$V1, HuFNSC02 = cortex02_GE02DN$V1, HuFNSC03 = cortex03_GE03DN$V1, HuFNSC04 = cortex04_GE04DN$V1)
# venn_DN_cortex_GE <- venn.diagram(DN_cortex_GE, filename = NULL, fill = c("red", "blue", "green", "yellow"), main = "Venn diagram of GE up-regulated genes")
plot.new()
grid.draw(venn_DN_cortex_GE)
```

```{r cortex_GE_enrich, fig.height = 8, fig.width = 8}
setwd("~/快盘/FetalBrain/RNAseq/DEfine/gene/")
# enrich_UP_cortex_GE <- enrich(name = "cortex_GE_UP_duplicated", fdr = 0.01, p = "FDR", erminej = F, height = 7, width = 9)
(enrich_UP_cortex_GE)
# enrich_DN_cortex_GE <- enrich(name = "cortex_GE_DN_duplicated", fdr = 0.01, p = "FDR", erminej = F, height = 7, width = 9)
(enrich_DN_cortex_GE)
# GOBP_UP_cortex_GE <- cbind(enrich_UP_cortex_GE$data, EX = "Cortex up-regulated")
# GOBP_UP_cortex_GE <- GOBP_UP_cortex_GE[as.character(GOBP_UP_cortex_GE$Category) == "GOBP", ]
# GOBP_UP_cortex_GE_plot <- ggplot(data = GOBP_UP_cortex_GE, aes(Term, -log10(FDR))) +
#   geom_bar(fill = "blue", stat = "identity", width = .5) + 
#   facet_grid(EX ~ .) + 
#   coord_flip() + 
#   geom_text(aes(label = round(-log10(FDR), 2), hjust = 0)) + 
#   xlab("") + 
#   theme_bw() +
#   scale_fill_hue(l = 40)
# (GOBP_UP_cortex_GE_plot + ggtitle("GOBP enrichment for Cortex up-regulated genes"))
# ggsave(GOBP_UP_cortex_GE_plot, file = "./enrich/GOBP_Cortex_GE.UP.pdf", height = 5, width = 6)
# GOBP_DN_cortex_GE <- cbind(enrich_DN_cortex_GE$data, EX = "GE up-regulated")
# GOBP_DN_cortex_GE <- GOBP_DN_cortex_GE[as.character(GOBP_DN_cortex_GE$Category) == "GOBP", ]
# GOBP_DN_cortex_GE_plot <- ggplot(data = GOBP_DN_cortex_GE, aes(Term, -log10(FDR))) +
#   geom_bar(fill = "blue", stat = "identity", width = .5) + 
#   facet_grid(EX ~ .) + 
#   coord_flip() + 
#   geom_text(aes(label = round(-log10(FDR), 2), hjust = 0)) + 
#   xlab("") + 
#   theme_bw() +
#   scale_fill_hue(l = 40)
# (GOBP_DN_cortex_GE_plot + ggtitle("GOBP enrichment for GE up-regulated genes"))
# ggsave(GOBP_DN_cortex_GE_plot, file = "./enrich/GOBP_Cortex_GE.DN.pdf", height = 5, width = 6)
```

```{r cortex_GE_shared, results='asis'}
# cortex_GE_UP_shared <- ensembl[intersect(intersect(cortex01_GE01UP$V1, cortex02_GE02UP$V1), intersect(cortex03_GE03UP$V1, cortex04_GE04UP$V1)), c("name", "description")]
cortex_GE_UP_shared$description <- gsub("_\\[.+", "", cortex_GE_UP_shared$description)
# cortex_GE_UP_shared <- data.frame(DE = "UP", cortex_GE_UP_shared)
cortex_GE_UP_shared <- xtable(cortex_GE_UP_shared)
align(cortex_GE_UP_shared) <- "ccrr"
print(cortex_GE_UP_shared, type = "html", include.rownames = F)

# cortex_GE_DN_shared <- ensembl[intersect(intersect(cortex01_GE01DN$V1, cortex02_GE02DN$V1), intersect(cortex03_GE03DN$V1, cortex04_GE04DN$V1)), c("name", "description")]
cortex_GE_DN_shared$description <- gsub("_\\[.+", "", cortex_GE_DN_shared$description)
# cortex_GE_DN_shared <- data.frame(DE = "DN", cortex_GE_DN_shared)
cortex_GE_DN_shared <- xtable(cortex_GE_DN_shared)
align(cortex_GE_DN_shared) <- "ccrr"
print(cortex_GE_DN_shared, type = "html", include.rownames = F)
```

### Between MZ twins - HuFNSC01 vs HuFNSC02

  * On average, there are __`r as.integer(mean(individual_DE_summary$DE[1:3]))`__ DE genes across three cells types.   
  * Majority of DE genes are cell type specific, only __`r sum(duplicated(c(rownames(brain01_brain02DE), rownames(cortex01_cortex02DE), rownames(GE01_GE02DE))))`__ are shared between any two cell types.   
  * DAVID enrichment analysis between MZ twins in brain and cortex show similar GO term in __brain development__, but there is no significantly enriched terms in GE.    

```{r individual, results='asis', fig.height = 6, fig.width = 6}
setwd("~/快盘/FetalBrain/RNAseq/DEfine/gene/")
individual_DE_summary <- xtable(individual_DE_summary)
align(individual_DE_summary) <- "lccc"
print(individual_DE_summary, type = "html", include.rownames = T)

# DE_HuFNSC01_HuFNSC02 <- list(brain = rownames(brain01_brain02DE), cortex = rownames(cortex01_cortex02DE), GE = rownames(GE01_GE02DE))
# venn_DE_HuFNSC01_HuFNSC02 <- venn.diagram(DE_HuFNSC01_HuFNSC02, filename = NULL, fill = c("green", "red", "blue"), main = "Venn diagram of HuFNSC01 vs HuFNSC02 DE genes", main.cex = 2)
# DE_HuFNSC03_HuFNSC04 <- list(cortex = rownames(cortex03_cortex04DE), GE = rownames(GE03_GE04DE))
# venn_DE_HuFNSC03_HuFNSC04 <- venn.diagram(DE_HuFNSC03_HuFNSC04, filename = NULL, fill = c("red", "blue"), main = "Venn diagram of HuFNSC03 vs HuFNSC04 DE genes", main.cex = 2)
plot.new()
grid.draw(venn_DE_HuFNSC01_HuFNSC02)
# plot.new()
# grid.draw(venn_DE_HuFNSC03_HuFNSC04)
```

```{r individual_enrich, fig.height = 4, fig.width = 8}
setwd("~/快盘/FetalBrain/RNAseq/DEfine/gene/")
# enrich_brain01_brain02DE <- enrich(name = "brain01_brain02DE", fdr = 0.01, p = "FDR", erminej = F, height = 4, width = 9)
(enrich_brain01_brain02DE)
# enrich_cortex01_cortex02DE <- enrich(name = "cortex01_cortex02DE", fdr = 0.01, p = "FDR", erminej = F, height = 4, width = 9)
(enrich_cortex01_cortex02DE)
# enrich_GE01_GE02DE <- enrich(name = "GE01_GE02DE", fdr = 0.01, p = "FDR", erminej = T, height = 6, width = 9)
# (enrich_GE01_GE02DE)
```

```{r individual_shared, results='asis'}
# individual_DE_shared <- ensembl[intersect(intersect(brain01_brain02DE$V1, cortex01_cortex02DE$V1), GE01_GE02DE$V1), c("name", "description")]
# individual_DE_shared$Brain <- brain01_brain02DE[rownames(individual_DE_shared), "DE"]
# individual_DE_shared$Cortex <- cortex01_cortex02DE[rownames(individual_DE_shared), "DE"]
# individual_DE_shared$GE <- GE01_GE02DE[rownames(individual_DE_shared), "DE"]
individual_DE_shared$description <- gsub("_\\[.+", "", individual_DE_shared$description)
individual_DE_shared <- xtable(individual_DE_shared)
align(individual_DE_shared) <- "crrccc"
print(individual_DE_shared, type = "html", include.rownames = F)
save(cortex_GE_DE_summary, cortex_GE_UP_duplicated, cortex_GE_DN_duplicated, cortex_GE_UP_shared, cortex_GE_DN_shared, cortex01_GE01DE, cortex02_GE02DE, cortex03_GE03DE, cortex04_GE04DE, venn_UP_cortex_GE, venn_DN_cortex_GE, enrich_UP_cortex_GE, enrich_DN_cortex_GE, individual_DE_summary, brain01_brain02DE, cortex01_cortex02DE, GE01_GE02DE, venn_DE_HuFNSC01_HuFNSC02, enrich_brain01_brain02DE, enrich_cortex01_cortex02DE, individual_DE_shared, file = "~/快盘/FetalBrain/RNAseq/DEfine/gene/FetalBrain_DEgenes.Rdata")
```




