---
output: 
  html_document:
    keep_md: true
    toc: yes
---
Fetal Brain MeDIP Analysis Summary
========================================================

Gloria Li         
`r date()` 

```{r include = FALSE}
library(knitr)
library(ggplot2)
library(xtable)
library(VennDiagram)
library(grid) 
load("~/快盘/FetalBrain/MeDIP/DMR/MZ/DMR_MZ.Rdata")
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

## DMR analysis from MeDIP fractional calls

  * DM CpG identification: $\Delta$mC < -0.4: hypomethylated; $\Delta$mC > 0.4: hypermethylated.       
  * Collapse DM CpGs into DMRs:   
    + adjacent CpGs have the same DM status;    
    + distance between adjacent CpGs $\le$ 200bp;   
    + No. of CpGs within each DMR $\ge$ 3.   
    
## DMRs between monozygptic twins
### Summary and sanity check

  * On average, there are __`r as.integer(mean(DMR_MZ_summary$Total))`__ DMR regions identified across three cell types, with __`r as.integer(mean(DMR_MZ_summary$Hyper))`__ hypermethylated, and __`r as.integer(mean(DMR_MZ_summary$Hypo))`__ hypomethylated.      
  * Median length of all DMRs is __`r median(DMR_pos_MZ_figure$data$length)`bp__. __Smaller__ than observed in breast (about half). _Bad script or biology?_      
  * Median No. of CpGs per DMR is __`r median(DMR_pos_MZ_figure$data$count)`__. Also __smaller__ than in breast.        
  * Median distance between adjacent DMRs is approximately __`r as.integer(median(DMR_pos_MZ_figure$data$dis[DMR_pos_MZ_figure$data$dis != -100000])/1000)`Kb__.         
  * DMRs are scattered along the genome, no obvious chunks of hyper/hypo regions.         

```{r DMR_MZ_sanity, results='asis', fig.width = 10, fig.height = 7}
load("~/快盘/FetalBrain/MeDIP/DMR/MZ/DMR_MZ.Rdata")
DMR_MZ_summary <- xtable(DMR_MZ_summary)
align(DMR_MZ_summary) <- "lccc"
print(DMR_MZ_summary, type = "html", include.rownames = T)

(DMR_length_MZ_figure + ggtitle("DMR length between monozygotic twins"))
(DMR_count_MZ_figure + ggtitle("No. of CpGs per DMR between monozygotic twins"))
(DMR_dis_MZ_figure + ggtitle("Distance between adjacent DMRs between monozygotic twins"))
(DMR_pos_MZ_figure + ggtitle("DMR positions on the chromosomes between monozygotic twins"))
```

### MZ DMR asymmetry

  + There is an __asymmetry__ between hyper/hypo DMRs across all three cell types, most significant in cortex neurosphere, and least so in GE neurosphere. Fold change in total DMR length hyper/hypo in Brain is __`r round(sum(DMR_pos_MZ_figure$data[DMR_pos_MZ_figure$data$cell == "Brain" & DMR_pos_MZ_figure$data$DM == 1, "length"])/sum(DMR_pos_MZ_figure$data[DMR_pos_MZ_figure$data$cell == "Brain" & DMR_pos_MZ_figure$data$DM == -1, "length"]), 3)`__, in Cortex is __`r round(sum(DMR_pos_MZ_figure$data[DMR_pos_MZ_figure$data$cell == "Cortex" & DMR_pos_MZ_figure$data$DM == 1, "length"])/sum(DMR_pos_MZ_figure$data[DMR_pos_MZ_figure$data$cell == "Cortex" & DMR_pos_MZ_figure$data$DM == -1, "length"]), 3)`__, and in GE is __`r round(sum(DMR_pos_MZ_figure$data[DMR_pos_MZ_figure$data$cell == "GE" & DMR_pos_MZ_figure$data$DM == 1, "length"])/sum(DMR_pos_MZ_figure$data[DMR_pos_MZ_figure$data$cell == "GE" & DMR_pos_MZ_figure$data$DM == -1, "length"]), 3)`__.    
  
```{r DMR_MZ_asymmetry, fig.width = 10, fig.height = 8}
(DMR_freq_MZ_figure + ggtitle("DMR frequency between monozygotic twins"))
```

### MZ DMR breakdown

  + Majority of DM CpGs overlap with genes, on average __`r round(mean(genomicBreak_MZ$Gene)*100, 2)`%__. And DM CpGs are __`r round(mean(genomicBreak_MZ$Promoter)/(3727169/28217448), 0)`-fold__ enriched in promoter regions, with on average __`r round(mean(genomicBreak_MZ_figure$data[genomicBreak_MZ_figure$data$Region == "Promoter", "CpG"])*100, 2)`%__ DM CpGs are in promoter regions.         
<!-- For the entire genome, 3727169 out of 28217448 CpGs overlap with TSS +/- 1500bp promoter regions -->    
```{r DMR_MZ_breakdown, fig.width = 7, fig.height = 5}
(genomicBreak_MZ_figure + ggtitle("DM CpG breakdown between monozygotic twins"))
```

### MZ DMR intersect with protein-coding promoters

  + On average, there are __`r as.integer(mean(DMR_gene_MZ_summary$pc.Promoters))`__ DMRs associated with promoters of protein-coding genes, __`r round(mean(DMR_gene_MZ_summary$pc.Promoters)/mean(DMR_MZ_summary$Total)*100, 2)`%__ of all DMRs, corresponds to __`r as.integer(mean(DMR_gene_MZ_summary$unique.Promoters))`__ protein-coding genes.         
  + These genes are __mostly shared__ by all three cell types.       
  + DAVID enrichment (FDR = 10^-6) shows these genes are enriched in __transcription regulation and neuron development__. 
  
```{r DMR_MZ_promoter, results='asis', fig.width = 8, fig.height = 8}
DMR_gene_MZ_summary <- xtable(DMR_gene_MZ_summary)
align(DMR_gene_MZ_summary) <- "lccccccc"
print(DMR_gene_MZ_summary, type = "html", include.rownames = T)

plot.new()
grid.draw(venn_DMR_pcPromoter_MZ)
(Brain01_Brain02_DMR_pcPromoter_DAVID)
(Cortex01_Cortex02_DMR_pcPromoter_DAVID)
(GE01_GE02_DMR_pcPromoter_DAVID)
```

### MZ DMR in promoters intersect with DE genes

  + On average, there are __`r as.integer(mean(DMR_gene_MZ_summary$proximal.DE.Genes))`__ promoter DMRs associated with DE genes, __`r round(mean(DMR_gene_MZ_summary$proximal.DE.Genes)/mean(DMR_gene_MZ_summary$pc.Promoters)*100, 2)`%__ of all promoter DMRs. Among them, there are __`r as.integer(mean(DMR_gene_MZ_summary$unique.DE.Genes))`__ unique DE genes associated with promoter DMRs across three cell types.         
  + Most of the DE genes are __cell type specific__.       
  + These DE genes are enriched in __neurogenesis and migration__ in both Brain and Cortex, but no enriched terms were found for GE.             

```{r DMR_MZ_DE, fig.width = 8, fig.height = 6}
plot.new()
grid.draw(venn_DMR_pcPromoter_DE_MZ)
(Brain01_Brain02_DMR_pcPromoter_DE_DAVID)
(Cortex01_Cortex02_DMR_pcPromoter_DE_DAVID)
```



