---
title: "FetalBrain - Figures"
author: "Gloria Li"
date: "November 4, 2014"
output: 
  html_document:
    keep_md: true
    toc: true
    toc_depth: 4
---

Updated: `r date()`

```{r setup, include = FALSE}
library(ggplot2)
library(plyr)
library(dplyr)
library(xtable)
library(VennDiagram)
library(gridExtra)
library(gplots)
library(dendextend)
library(reshape)
library(wq)
load("~/快盘/FetalBrain/MeDIP/DMR_MZ.Rdata")
load("~/快盘/FetalBrain/MeDIP/DMR_neurospheres.Rdata")
load("~/快盘/FetalBrain/WGBS/WGBS.DMR.Rdata")
load("~/快盘/FetalBrain/RNAseq/DEfine/gene/FetalBrain_DEgenes.Rdata")
load("~/快盘/FetalBrain/RNAseq/isoform/FetalBrain_isoform.Rdata")
load("~/快盘/FetalBrain/RNAseq/junction/FetalBrain_isoform_valid.Rdata")
load("~/快盘/FetalBrain/RNAseq/epiProfile/exonProfile_Rmd.Rdata")
load("~/快盘/FetalBrain/miR/FetalBrain_miR.Rdata")
load("~/快盘/FetalBrain/GW/GW.Rdata")
knitr::opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE, results = FALSE, fig.height = 12, fig.width = 12)
```

## Figure 1: Summary   
### Figure 1a: Experimental design overview   

![](./Figures_files/figure-html/design.png)     

### Figure 1b: Comparisons setup  

![](./Figures_files/figure-html/comparisons.png)     
    
### Figure 1c: UMRs and DE genes between MZ twins and neurospheres 
  * No. of UMRs (hypermethylated in red and hypomethylated in blue) between monozygotic twins and Cortex vs GE neurospheres.    
  * No. of UMRs between MZ twins are on the same scale as UMRs between neurospheres.   
  
```{r MZ_scale, fig.height = 10, fig.width = 12}
MZ_scale <- data.frame(Sample = rep(c(DMR_MZ_summary$Sample, DMR_neurospheres_summary$Sample), 5), 
                        Category = rep(c(rep("Individual specific", nrow(DMR_MZ_summary)), rep("Cell type specific", nrow(DMR_neurospheres_summary))), 5), 
                        DM_DE = c(rep(c("Hyper / UP", "Hypo / DN"), each = nrow(DMR_MZ_summary) + nrow(DMR_neurospheres_summary)), rep(c("Hyper / UP", "Hypo / DN"), each = nrow(DMR_MZ_summary) + nrow(DMR_neurospheres_summary)), rep("Isoforms", nrow(DMR_MZ_summary) + nrow(DMR_neurospheres_summary))), 
                        Assay = factor(c(rep(c("UMRs", "DE genes"), each = (nrow(DMR_MZ_summary) + nrow(DMR_neurospheres_summary))*2), rep("Isoforms", (nrow(DMR_MZ_summary) + nrow(DMR_neurospheres_summary)))), levels = c("UMRs", "DE genes", "Isoforms")), 
                        value = c((filter(DMR_freq_MZ_figure$data, DM == 1) %>% group_by(cell) %>% summarize(m = mean(freq)))$m, (filter(DMR_freq_neurospheres_figure$data, DM == 1) %>% group_by(donor) %>% summarize(m = mean(freq)))$m, (filter(DMR_freq_MZ_figure$data, DM == -1) %>% group_by(cell) %>% summarize(m = mean(freq)))$m, (filter(DMR_freq_neurospheres_figure$data, DM == -1) %>% group_by(donor) %>% summarize(m = mean(freq)))$m, individual_DE_summary$UP[1:3], cortex_GE_DE_summary$UP[1:2], -individual_DE_summary$DN[1:3], -cortex_GE_DE_summary$DN[1:2], individual_summary[1:3, "isoform_genes"], cortex_GE_summary[1:2, "isoform_genes"]))
MZ_scale$Sample <- factor(gsub("-HuFNSC", "", MZ_scale$Sample), levels = c("Brain01_Brain02", "Cortex01_Cortex02", "GE01_GE02", "Cortex01_GE01", "Cortex02_GE02"))
MZ_scale$Category <- factor(MZ_scale$Category, levels = c("Individual specific", "Cell type specific"))
MZ_scale_figure <- ggplot(MZ_scale, aes(x = Sample, y = value, fill = DM_DE)) + 
   geom_bar(stat="identity", position = "identity", width = 0.5) + 
   geom_hline(yintercept = 0) + 
   facet_grid(Assay ~ Category, scales = "free", space = "free_x") + 
   scale_fill_manual(values = c("red", "blue", "black"), name = "") + 
   scale_y_continuous(breaks = c(-3000, -2000, -500, -100, 0, 100, 500, 2000, 3000), labels = c(3000, 2000, 500, 100, 0, 100, 500, 2000, 3000)) + 
   xlab("") + 
   ylab("") + 
   ggtitle("MZ twins") + 
   theme_bw() + 
   theme(strip.text = element_text(size = 15), axis.title.y = element_text(size = 15), axis.text = element_text(size = 12), legend.text = element_text(size = 15))
MZ_scale_figure_gt <- ggplot_gtable(ggplot_build(MZ_scale_figure))
grid.newpage()
grid.draw(MZ_scale_figure_gt)
grid.text("UMR frequency", x = unit(0.015, "npc"), y = unit(0.8, "npc"), rot = 90, gp=gpar(fontsize=15))
grid.text("No. of genes", x = unit(0.015, "npc"), y = unit(0.35, "npc"), rot = 90, gp=gpar(fontsize=15))
pdf("~/快盘/FetalBrain/Figures/Figure1c.pdf", height = 10, width = 12)
grid.newpage()
grid.draw(MZ_scale_figure_gt)
grid.text("UMR frequency", x = unit(0.015, "npc"), y = unit(0.8, "npc"), rot = 90, gp=gpar(fontsize=15))
grid.text("No. of genes", x = unit(0.015, "npc"), y = unit(0.35, "npc"), rot = 90, gp=gpar(fontsize=15))
dev.off()
```

### Figure 1d: UMRs and DE genes between neurospheres accumulates over time

```{r neurospheres_scale, fig.height = 10, fig.width = 12}
neurospheres_scale <- data.frame(Donor = c(rep(c("HuFNSC02", "HuFNSC04", "HuFNSC01", "HuFNSC02", "HuFNSC03", "HuFNSC04"), 2), "HuFNSC01", "HuFNSC02", "HuFNSC03", "HuFNSC04"), 
                                 GW = c(rep(c("GW17", "GW13", "GW17", "GW17", "GW13", "GW13"), 2), "GW17", "GW17", "GW13", "GW13"), 
                                 Assay = factor(c(rep(c(rep("UMRs", 2), rep("DE genes", 4)), 2), rep("Isoforms", 4)), levels = c("UMRs", "DE genes", "Isoforms")), 
                                 DM_DE = c(rep(c("Cortex UMR/UP", "GE UMR/UP"), each = 6), rep("Isoform", 4)), 
                                 value = c((filter(DMR_freq_WGBS_figure$data, DM == -1) %>% group_by(donor) %>% summarize(m = -mean(freq)))$m, cortex_GE_DE_summary$UP, (filter(DMR_freq_WGBS_figure$data, DM == 1) %>% group_by(donor) %>% summarize(m = mean(freq)))$m, cortex_GE_DE_summary$DN, cortex_GE_summary[, "isoform_genes"]))
neurospheres_scale[neurospheres_scale$DM_DE == "GE UMR/UP", "value"] <- -neurospheres_scale[neurospheres_scale$DM_DE == "GE UMR/UP", "value"]
neurospheres_scale <- filter(neurospheres_scale, Donor %in% c("HuFNSC02", "HuFNSC04"))
neurospheres_scale_figure <- ggplot(neurospheres_scale, aes(x = Donor, y = value, fill = DM_DE)) + 
   geom_bar(stat="identity", position = "identity", width = 0.4) + 
   geom_hline(yintercept = 0) + 
   facet_grid(Assay ~ GW, scales = "free") + 
   scale_fill_manual(values = c("red", "blue", "black"), name = "") + 
   scale_y_continuous(breaks = c(-400, -200, -50, 0, 50, 200, 400, 1000, 2000), labels = abs(c(-400, -200, -50, 0, 50, 200, 400, 1000, 2000))) + 
   xlab("") + 
   ylab("") + 
   ggtitle("Neurospheres") + 
   theme_bw() + 
   theme(strip.text = element_text(size = 15), axis.title.y = element_text(size = 15), axis.text = element_text(size = 12), legend.text = element_text(size = 15))
neurospheres_scale_figure_gt <- ggplot_gtable(ggplot_build(neurospheres_scale_figure))
grid.newpage()
grid.draw(neurospheres_scale_figure_gt)
grid.text("UMR frequency", x = unit(0.015, "npc"), y = unit(0.8, "npc"), rot = 90, gp=gpar(fontsize=15))
grid.text("No. of genes", x = unit(0.015, "npc"), y = unit(0.35, "npc"), rot = 90, gp=gpar(fontsize=15))
pdf("~/快盘/FetalBrain/Figures/Figure1d.pdf", height = 10, width = 12)
grid.newpage()
grid.draw(neurospheres_scale_figure_gt)
grid.text("UMR frequency", x = unit(0.015, "npc"), y = unit(0.8, "npc"), rot = 90, gp=gpar(fontsize=15))
grid.text("No. of genes", x = unit(0.015, "npc"), y = unit(0.35, "npc"), rot = 90, gp=gpar(fontsize=15))
dev.off()
```

### Figure 1e: UMRs and DE genes between GW 

```{r GW_scale, fig.height = 10, fig.width = 12}
GW_scale <- data.frame(Cell = rep(c("Cortex", "GE"), 5), 
                       Assay = factor(c(rep(c(rep("UMRs", 2), rep("DE genes", 2)), 2), rep("Isoforms", 2)), levels = c("UMRs", "DE genes", "Isoforms")), 
                       DM_DE = c(rep(c("GW17 UMR/UP", "GW13 UMR/UP"), each = 4), rep("Isoform", 2)), 
                       value = c((filter(DMR_freq_GW_figure$data, DM == -1) %>% group_by(cell) %>% summarize(m = -mean(freq)))$m, GW_DE_summary[GW_DE_summary$Sample %in% c("GW17-GW13.2_cortex", "GW17-GW13.2_GE"), ]$UP, (filter(DMR_freq_GW_figure$data, DM == 1) %>% group_by(cell) %>% summarize(m = mean(freq)))$m, GW_DE_summary[GW_DE_summary$Sample %in% c("GW17-GW13.2_cortex", "GW17-GW13.2_GE"), ]$DN, as.numeric(as.vector(GW_isoform_summary[GW_isoform_summary$Sample %in% c("Cortex-HuFNSC02_Cortex-HuFNSC04", "GE-HuFNSC02_GE-HuFNSC04"), "isoform_genes"]))))
GW_scale[GW_scale$DM_DE == "GW13 UMR/UP", "value"] <- -GW_scale[GW_scale$DM_DE == "GW13 UMR/UP", "value"]
GW_scale_figure <- ggplot(GW_scale, aes(x = Cell, y = value, fill = DM_DE)) + 
   geom_bar(stat="identity", position = "identity", width = 0.4) + 
   geom_hline(yintercept = 0) + 
   facet_grid(Assay ~ Cell, scales = "free") + 
   scale_fill_manual(values = c("red", "blue", "black"), name = "") + 
   scale_y_continuous(breaks = c(-200, -100, 0, 100, 250, 500, 1000, 1500, 2000), labels = abs(c(-200, -100, 0, 100, 250, 500, 1000, 1500, 2000))) + 
   xlab("") + 
   ylab("") + 
   ggtitle("GW") + 
   theme_bw() + 
   theme(strip.text = element_text(size = 15), axis.title.y = element_text(size = 15), axis.text = element_text(size = 12), legend.text = element_text(size = 15))
GW_scale_figure_gt <- ggplot_gtable(ggplot_build(GW_scale_figure))
grid.newpage()
grid.draw(GW_scale_figure_gt)
grid.text("UMR frequency", x = unit(0.015, "npc"), y = unit(0.8, "npc"), rot = 90, gp=gpar(fontsize=15))
grid.text("No. of genes", x = unit(0.015, "npc"), y = unit(0.35, "npc"), rot = 90, gp=gpar(fontsize=15))
pdf("~/快盘/FetalBrain/Figures/Figure1e.pdf", height = 10, width = 12)
grid.newpage()
grid.draw(GW_scale_figure_gt)
grid.text("UMR frequency", x = unit(0.015, "npc"), y = unit(0.8, "npc"), rot = 90, gp=gpar(fontsize=15))
grid.text("No. of genes", x = unit(0.015, "npc"), y = unit(0.35, "npc"), rot = 90, gp=gpar(fontsize=15))
dev.off()
```

## Figure 2: Neurospheres Cortex and GE
### Figure 2a: TFBS asymmetry between neurosphere UMRs

```{r neurospheres_TFBS}
(DMR_TF_figure <- DMR_TF_figure + xlab("Abundance near Cortex UMRs") + ylab("Abundance near GE UMRs")) 
ggsave(DMR_TF_figure, file = "~/快盘/FetalBrain/Figures/Figure2a.pdf")
```

### Figure 2b: GREAT enrichment for neurospheres UMRs

```{r neurospheres_GREAT}
GOBP_Cortex02.UMR_WGBS <- read.delim("~/快盘/FetalBrain/WGBS/DMR/enrich/ALL_GREAT_GOBP_DMR.Cortex-HuFNSC02_GE-HuFNSC02.hypo.tsv", head = F, as.is = T, skip = 4)
GOBP_GE02.UMR_WGBS <- read.delim("~/快盘/FetalBrain/WGBS/DMR/enrich/ALL_GREAT_GOBP_DMR.Cortex-HuFNSC02_GE-HuFNSC02.hyper.tsv", head = F, as.is = T, skip = 4)
GOBP_Cortex02_GE02_WGBS <- rbind(select(mutate(filter(GOBP_Cortex02.UMR_WGBS, V8 >= 2 & V7 <= 0.05 & V16 <= 0.05), UMR = "Cortex.UMR", FDR = -log10(V7), Term = V3), UMR, Term, FDR),
                                 select(mutate(filter(GOBP_GE02.UMR_WGBS, V8 >= 2 & V7 <= 0.05 & V16 <= 0.05), UMR = "GE.UMR", FDR = log10(V7), Term = V3), UMR, Term, FDR))
Terms <- unique(c(arrange(filter(GOBP_Cortex02_GE02_WGBS, UMR == "Cortex.UMR"), desc(FDR))[1:20, "Term"], rev(arrange(filter(GOBP_Cortex02_GE02_WGBS, UMR == "GE.UMR"), FDR)[1:20, "Term"])))
GOBP_Cortex02_GE02_WGBS <- mutate(filter(GOBP_Cortex02_GE02_WGBS, Term %in% Terms), Term = factor(Term, levels = rev(Terms)))
(GOBP_Cortex02_GE02_WGBS_figure <- ggplot(data = GOBP_Cortex02_GE02_WGBS, aes(Term, FDR, fill = UMR)) +
  geom_bar(position = "identity", stat = "identity", width = .5) + 
  coord_flip() + 
  xlab("") + 
  ylab("-log10(FDR)") + 
  scale_fill_manual(values = c("blue", "red"), name = "") + 
  theme_bw())
ggsave(GOBP_Cortex02_GE02_WGBS_figure, file = "~/快盘/FetalBrain/Figures/Figure2b.pdf")
```

## Figure 3: Stage-specific differential expression  
### Figure 3a: Stage-specific differential expression summary  

```{r GW_DE_summary}
(GW_DE_summary_figure)
ggsave(GW_DE_summary_figure, file = "~/快盘/FetalBrain/Figures/Figure3a.pdf")
```

### Figure 3b: Stage-specific differential expression trend - shared

```{r GW_DE_trend}
(GW_DE_trend_figure <- GW_DE_trend_figure + ylab("Relative expression level"))
ggsave(GW_DE_trend_figure, file = "~/快盘/FetalBrain/Figures/Figure3b.pdf")
```

### Figure 3c: Stage-specific differential expression trend - cortex

```{r GW_DE_trend_cortex}
(GW_DE_trend_cortex_figure <- GW_DE_trend_cortex_figure + ylab("Relative expression level"))
ggsave(GW_DE_trend_cortex_figure, file = "~/快盘/FetalBrain/Figures/Figure3c.pdf")
```

### Figure 3d: Stage-specific differential expression trend - GE

```{r GW_DE_trend_GE}
(GW_DE_trend_GE_figure <- GW_DE_trend_GE_figure + ylab("Relative expression level"))
ggsave(GW_DE_trend_GE_figure, file = "~/快盘/FetalBrain/Figures/Figure3d.pdf")
```

### Figure 3e: Stage-specific differential expression heatmap  

```{r GW_DE_heatmap}
clab <- data.frame(sample = factor(colnames(GW_DE_rpkm), levels = colnames(GW_DE_rpkm)),
                   y = 1, 
                   GW = rep(c("GW13", "GW15", "GW17", "GW17"), 2))
clab_figure <- ggplot(clab, aes(sample, y, fill = GW)) + 
   geom_tile() + 
   scale_fill_manual(values = c(rgb(250,192,144,maxColorValue = 255), rgb(247,150,70,maxColorValue = 255), rgb(228,108,10,maxColorValue = 255)), name = "", guide = guide_legend(nrow = 1)) + 
   xlab("") + 
   ylab("") + 
   theme_bw() + 
   theme(axis.text = element_text(size = 0), axis.ticks = element_line(color = "white"), panel.border = element_rect(color = "white"), plot.margin=unit(c(1,1,-0.3,1), "cm"), legend.position = "top", panel.grid = element_line(color = "white"))
GW_DE_heatmap <- ggplot(melt(t(scale(t(GW_DE_rpkm), center = T, scale = T))) %>% mutate(miR = factor(X1, levels = rev(rownames(GW_DE_rpkm))), Sample = factor(X2, levels = colnames(GW_DE_rpkm))), aes(x=Sample, y=miR, fill=value)) + 
   geom_tile() + 
   geom_hline(yintercept=as.numeric(factor(c("ENSG00000086570", "ENSG00000164929", "ENSG00000154736", "ENSG00000182580"), levels = rev(rownames(GW_DE_rpkm))))) + 
   scale_fill_gradient(name = "Z-score") + 
   xlab("") + 
   ylab("") + 
   theme_bw() + 
   theme(axis.text.y = element_text(size = 0), axis.ticks.y = element_line(color = "white"), panel.border = element_rect(color = "white"), axis.text.x = element_text(angle = 90), plot.margin=unit(c(-0.3,1,1,1), "cm"), panel.background = element_rect(fill = "white"), panel.grid = element_line(color = "white"), legend.position = "bottom")
grid.arrange(clab_figure, GW_DE_heatmap, nrow = 2, heights = c(0.12, 0.88))
pdf("~/快盘/FetalBrain/Figures/Figure3e.pdf", height = 12, width = 12)
grid.arrange(clab_figure, GW_DE_heatmap, nrow = 2, heights = c(0.12, 0.88))
dev.off()
```

## Supplemental Figures
### Supplemental 1: Asymmetry in UMRs 
  
```{r UMR_asymmetry}
grid.newpage()
layOut(list(DMR_freq_MZ_figure + ggtitle("UMRs between MZ twins"), 1, 1:3), 
       list(DMR_freq_WGBS_figure + ggtitle("UMRs between neurospheres"), 2, 1:2), 
       list(DMR_freq_GW_figure + ggtitle("UMRs between GW") + scale_fill_manual(values = c("red", "blue"), labels = c("GW13 UMRs", "GW17 UMRs"), name = ""), 2, 3:4))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.515, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf(file = "~/快盘/FetalBrain/Figures/FigureS1.pdf", height = 12, width = 12)
layOut(list(DMR_freq_MZ_figure + ggtitle("UMRs between MZ twins"), 1, 1:3), 
       list(DMR_freq_WGBS_figure + ggtitle("UMRs between neurospheres"), 2, 1:2), 
       list(DMR_freq_GW_figure + ggtitle("UMRs between GW") + scale_fill_manual(values = c("red", "blue"), labels = c("GW13 UMRs", "GW17 UMRs"), name = ""), 2, 3:4))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.515, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 2: Genomic breakdown of UMRs

```{r genomicBreak}
genomicBreak_MZ_tall <- genomicBreak_MZ_figure$data %>% mutate(DM = revalue(factor(DM), c("hyper" = "HuFNSC02 UMRs", "hypo" = "HuFNSC01 UMRs")))
genomicBreak_MZ_figure <- ggplot(genomicBreak_MZ_tall, aes(x = Region, y = log2(FC), fill = Sample)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
  xlab("") + 
  ylab("log2 Fold enrichment") + 
  ggtitle("UMRs between MZ twins") + 
  facet_wrap(~ DM) + 
  coord_flip() + 
  scale_fill_manual(values = c("green", "red", "blue"), labels = c("Brain", "Cortex", "GE"), name = "") + 
  theme_bw()
genomicBreak_WGBS_tall <- genomicBreak_WGBS_figure$data %>% mutate(DM = revalue(factor(DM), c("hyper" = "GE UMRs", "hypo" = "Cortex UMRs")))
genomicBreak_WGBS_figure <- ggplot(genomicBreak_WGBS_tall, aes(x = Region, y = log2(FC), fill = Sample)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
  xlab("") + 
  ylab("log2 Fold enrichment") + 
  ggtitle("UMRs between neurospheres") + 
  coord_flip() + 
  scale_fill_manual(values = c("red", "blue"), labels = c("HuFNSC02\n(GW17)", "HuFNSC04\n(GW13)"), name = "") + 
  facet_wrap(~ DM) + 
  theme_bw()
genomicBreak_GW_tall <- genomicBreak_GW_figure$data %>% mutate(DM = revalue(factor(DM), c("hyper" = "GW13 UMRs", "hypo" = "GW17 UMRs")))
genomicBreak_GW_figure <- ggplot(genomicBreak_GW_tall, aes(x = Region, y = log2(FC), fill = Sample)) + 
  geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
  xlab("") + 
  ylab("log2 Fold enrichment") + 
  ggtitle("UMRs between GW") + 
  coord_flip() + 
  scale_fill_manual(values = c("red", "blue"), labels = c("Cortex", "GE"), name = "") + 
  facet_wrap(~ DM) + 
  theme_bw()
grid.newpage()
layOut(list(genomicBreak_MZ_figure, 1, 1), 
       list(genomicBreak_WGBS_figure, 1, 2), 
       list(genomicBreak_GW_figure, 2, 1))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS2.pdf", height = 12, width = 12)
layOut(list(genomicBreak_MZ_figure, 1, 1), 
       list(genomicBreak_WGBS_figure, 1, 2), 
       list(genomicBreak_GW_figure, 2, 1))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 3: Fraction of DE genes with proximal UMRs

```{r proximal_DE}
DMR_proximal_DE_MZ_summary <- data.frame(Sample = rownames(DMR_proximal_MZ_summary), Total.UP = rep(c(nrow(filter(brain01_brain02DE, DE == "UP")), nrow(filter(cortex01_cortex02DE, DE == "UP")), nrow(filter(GE01_GE02DE, DE == "UP"))), each = 2), Total.DN = rep(c(nrow(filter(brain01_brain02DE, DE == "DN")), nrow(filter(cortex01_cortex02DE, DE == "DN")), nrow(filter(GE01_GE02DE, DE == "DN"))), each = 2), DMR.DE = DMR_proximal_MZ_summary[, "DE.DMRs"], same.direction = DMR_proximal_MZ_summary[, "same.direction"], DMR.UP = NA, DMR.DN = NA)
DMR_proximal_DE_MZ_summary$DM <- gsub(".*_", "", DMR_proximal_DE_MZ_summary$Sample)
DMR_proximal_DE_MZ_summary$Sample <- gsub("\\d.*", "", DMR_proximal_DE_MZ_summary$Sample)
DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "DMR.UP"] <- (DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "DMR.DE"] - DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "same.direction"]) / DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "Total.UP"]
DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "DMR.UP"] <- DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "same.direction"] / DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "Total.UP"]
DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "DMR.DN"] <- DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "same.direction"] / DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hyper", "Total.DN"]
DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "DMR.DN"] <- (DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "DMR.DE"] - DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "same.direction"])  / DMR_proximal_DE_MZ_summary[DMR_proximal_DE_MZ_summary$DM == "hypo", "Total.DN"]
DMR_proximal_DE_MZ_summary_tall <- data.frame(Sample = rep(DMR_proximal_DE_MZ_summary$Sample, 2), DM = rep(DMR_proximal_DE_MZ_summary$DM, 2), DE = rep(c("UP", "DN"), each = nrow(DMR_proximal_DE_MZ_summary)), Percent = c(DMR_proximal_DE_MZ_summary$DMR.UP, DMR_proximal_DE_MZ_summary$DMR.DN))
DMR_proximal_DE_MZ_summary_tall <- DMR_proximal_DE_MZ_summary_tall %>% mutate(DM = revalue(factor(DM), c("hyper" = "HuFNSC02 UMRs", "hypo" = "HuFNSC01 UMRs")))
DMR_proximal_DE_MZ_figure <- ggplot(DMR_proximal_DE_MZ_summary_tall, aes(x = Sample, y = Percent, fill = DE)) + 
   geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
   facet_wrap(~ DM) + 
   coord_flip() + 
   scale_fill_manual(values = c("blue", "red"), labels = c("UP in HuFNSC02", "UP in HuFNSC01"), name = "") + 
   xlab("") + 
   ylab("Fraction of DE genes") + 
   ggtitle("UMRs between MZ twins") + 
   theme_bw()
DMR_proximal_DE_neurospheres_summary <- data.frame(Sample = rownames(DMR_WGBS_gene_summary), Total.UP = rep(c(nrow(filter(cortex02_GE02DE, DE == "UP")), nrow(filter(cortex04_GE04DE, DE == "UP"))), each = 2), Total.DN = rep(c(nrow(filter(cortex02_GE02DE, DE == "DN")), nrow(filter(cortex04_GE04DE, DE == "DN"))), each = 2), DMR.DE = DMR_WGBS_gene_summary[, "proximal.DE.Genes"], same.direction = DMR_WGBS_gene_summary[, "same.direction"], DMR.UP = NA, DMR.DN = NA)
DMR_proximal_DE_neurospheres_summary$DM <- gsub("-.*", "", DMR_proximal_DE_neurospheres_summary$Sample)
DMR_proximal_DE_neurospheres_summary$Sample <- gsub(".*-", "", DMR_proximal_DE_neurospheres_summary$Sample)
DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "DMR.UP"] <- (DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "DMR.DE"] - DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "same.direction"]) / DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "Total.UP"]
DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "DMR.UP"] <- DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "same.direction"] / DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "Total.UP"]
DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "DMR.DN"] <- DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "same.direction"] / DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "GE_UMRs", "Total.DN"]
DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "DMR.DN"] <- (DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "DMR.DE"] - DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "same.direction"])  / DMR_proximal_DE_neurospheres_summary[DMR_proximal_DE_neurospheres_summary$DM == "Cortex_UMRs", "Total.DN"]
DMR_proximal_DE_neurospheres_summary$DM <- gsub("_", " ", DMR_proximal_DE_neurospheres_summary$DM)
DMR_proximal_DE_neurospheres_summary_tall <- data.frame(Sample = rep(DMR_proximal_DE_neurospheres_summary$Sample, 2), DM = rep(DMR_proximal_DE_neurospheres_summary$DM, 2), DE = rep(c("UP", "DN"), each = nrow(DMR_proximal_DE_neurospheres_summary)), Percent = c(DMR_proximal_DE_neurospheres_summary$DMR.UP, DMR_proximal_DE_neurospheres_summary$DMR.DN))
DMR_proximal_DE_neurospheres_figure <- ggplot(DMR_proximal_DE_neurospheres_summary_tall, aes(x = Sample, y = Percent, fill = DE)) + 
   geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
   facet_wrap(~ DM) + 
   coord_flip() + 
   scale_fill_manual(values = c("blue", "red"), labels = c("UP in GE", "UP in Cortex"), name = "") + 
   scale_y_continuous(breaks = c(0.01, 0.02)) + 
   xlab("") + 
   ylab("Fraction of DE genes") + 
   ggtitle("UMRs between neurospheres") + 
   theme_bw()
DMR_proximal_DE_GW_summary <- data.frame(Sample = rownames(DMR_proximal_GW_summary), Total.UP = rep(c(nrow(filter(cortex02_cortex04DE, DE == "UP")), nrow(filter(GE02_GE04DE, DE == "UP"))), each = 2), Total.DN = rep(c(nrow(filter(cortex02_cortex04DE, DE == "DN")), nrow(filter(GE02_GE04DE, DE == "DN"))), each = 2), DMR.DE = DMR_proximal_GW_summary[, "DE.DMRs"], same.direction = DMR_proximal_GW_summary[, "same.direction"], DMR.UP = NA, DMR.DN = NA)
DMR_proximal_DE_GW_summary$DM <- c("hyper", "hypo", "hyper", "hypo")
DMR_proximal_DE_GW_summary$Sample <- gsub("\\d.*", "", DMR_proximal_DE_GW_summary$Sample)
DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "DMR.UP"] <- (DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "DMR.DE"] - DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "same.direction"]) / DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "Total.UP"]
DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "DMR.UP"] <- DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "same.direction"] / DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "Total.UP"]
DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "DMR.DN"] <- DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "same.direction"] / DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hyper", "Total.DN"]
DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "DMR.DN"] <- (DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "DMR.DE"] - DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "same.direction"])  / DMR_proximal_DE_GW_summary[DMR_proximal_DE_GW_summary$DM == "hypo", "Total.DN"]
DMR_proximal_DE_GW_summary_tall <- data.frame(Sample = rep(DMR_proximal_DE_GW_summary$Sample, 2), DM = rep(DMR_proximal_DE_GW_summary$DM, 2), DE = rep(c("UP", "DN"), each = nrow(DMR_proximal_DE_GW_summary)), Percent = c(DMR_proximal_DE_GW_summary$DMR.UP, DMR_proximal_DE_GW_summary$DMR.DN))
DMR_proximal_DE_GW_summary_tall <- DMR_proximal_DE_GW_summary_tall %>% mutate(DM = revalue(factor(DM), c("hyper" = "GW13 UMRs", "hypo" = "GW17 UMRs")))
DMR_proximal_DE_GW_figure <- ggplot(DMR_proximal_DE_GW_summary_tall, aes(x = Sample, y = Percent, fill = DE)) + 
   geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
   facet_wrap(~ DM) + 
   coord_flip() + 
   scale_fill_manual(values = c("red", "blue"), labels = c("UP in GW13", "UP in GW17"), name = "") + 
   xlab("") + 
   ylab("Fraction of DE genes") + 
   ggtitle("UMRs between GW") + 
   theme_bw()
grid.newpage()
layOut(list(DMR_proximal_DE_MZ_figure, 1, 1), 
       list(DMR_proximal_DE_neurospheres_figure, 1, 2), 
       list(DMR_proximal_DE_GW_figure, 2, 1))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS3.pdf", height = 12, width = 12)
layOut(list(DMR_proximal_DE_MZ_figure, 1, 1), 
       list(DMR_proximal_DE_neurospheres_figure, 1, 2), 
       list(DMR_proximal_DE_GW_figure, 2, 1))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 4: Venn diagram of differential expressed genes between MZ and between GW 
  * No. of differential expressed genes between MZ twins in Brain (green), Neurospheres Cortex derived (red), and GE derived (blue).  
  
```{r Venn_DE}
MZ_UP <- list(Brain = as.character(filter(brain01_brain02DE, DE == "UP")[, "V1"]), Cortex = as.character(filter(cortex01_cortex02DE, DE == "UP")[, "V1"]), GE = as.character(filter(GE01_GE02DE, DE == "UP")[, "V1"]))
venn_MZ_UP <- venn.diagram(MZ_UP, filename = NULL, fill = c("green", "red", "blue"), main = "MZ DE UP")
MZ_DN <- list(Brain = as.character(filter(brain01_brain02DE, DE == "DN")[, "V1"]), Cortex = as.character(filter(cortex01_cortex02DE, DE == "DN")[, "V1"]), GE = as.character(filter(GE01_GE02DE, DE == "DN")[, "V1"]))
venn_MZ_DN <- venn.diagram(MZ_DN, filename = NULL, fill = c("green", "red", "blue"), main = "MZ DE DN")
GW_cortex <- list(GW17_GW13 = c(GW_17_13_UP_duplicated_cortex$id, GW_17_13_DN_duplicated_cortex$id), GW17_GW15 = c(GW_17_15_UP_duplicated_cortex$id, GW_17_15_DN_duplicated_cortex$id), GW15_GW13 = cortex03_cortex04DE$ID)
venn_GW_cortex <- venn.diagram(GW_cortex, filename = NULL, fill = c("red", "blue", "green"), main = "GW DE in cortex", category.names = c("17 vs 13", "15 vs 13", "17 vs 15"))
GW_GE <- list(GW17_GW13 = c(GW_17_13_UP_duplicated_GE$id, GW_17_13_DN_duplicated_GE$id), GW17_GW15 = c(GW_17_15_UP_duplicated_GE$id, GW_17_15_DN_duplicated_GE$id), GW15_GW13 = GE03_GE04DE$ID)
venn_GW_GE <- venn.diagram(GW_GE, filename = NULL, fill = c("red", "blue", "green"), main = "GW DE in GE", category.names = c("17 vs 13", "15 vs 13", "17 vs 15"))
grid.arrange(gTree(children = venn_MZ_UP), gTree(children = venn_MZ_DN), gTree(children = venn_GW_cortex), gTree(children = venn_GW_GE), nrow = 2)
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS4.pdf", height = 12, width = 12)
grid.arrange(gTree(children = venn_MZ_UP), gTree(children = venn_MZ_DN), gTree(children = venn_GW_cortex), gTree(children = venn_GW_GE), nrow = 2)
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 5: DAVID enrichment for MZ DE genes

```{r DE_DAVID}
DAVID_brain01_brain02DE <- read.delim("~/快盘/FetalBrain/RNAseq/DEfine/gene/enrich/brain01_brain02DE_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
DAVID_cortex01_cortex02DE <- read.delim("~/快盘/FetalBrain/RNAseq/DEfine/gene/enrich/cortex01_cortex02DE_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
DAVID_GE01_GE02DE <- read.delim("~/快盘/FetalBrain/RNAseq/DEfine/gene/enrich/GE01_GE02DE_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
Terms <- unique(c(DAVID_brain01_brain02DE[DAVID_brain01_brain02DE$FDR < 0.01, "Term"], DAVID_cortex01_cortex02DE[DAVID_cortex01_cortex02DE$FDR < 0.01, "Term"], DAVID_GE01_GE02DE[DAVID_GE01_GE02DE$FDR < 0.01, "Term"]))
DAVID_MZ <- rbind(filter(DAVID_brain01_brain02DE, Term %in% Terms) %>% mutate(Cell = "Brain"), filter(DAVID_cortex01_cortex02DE, Term %in% Terms) %>% mutate(Cell = "Cortex"), filter(DAVID_GE01_GE02DE, Term %in% Terms) %>% mutate(Cell = "GE")) %>% mutate(Term = gsub("GO:\\d+~", "", Term), FDR = -log10(FDR) -2) %>% select(Cell, Term, FDR)
DAVID_MZ$Term <- factor(DAVID_MZ$Term, levels = rev(unique(DAVID_MZ[order(DAVID_MZ$FDR, decreasing = T), "Term"])))
DAVID_MZ_figure <- ggplot(DAVID_MZ, aes(x = Term, y = FDR, fill = Cell)) + 
   geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
   coord_flip() + 
   scale_fill_manual(values = c("green", "red", "blue"), labels = c("Brain", "Cortex", "GE"), name = "") + 
   scale_y_continuous(breaks = seq(-4, 6, by = 2), labels = seq(-2, 8, by = 2)) + 
   xlab("") + 
   ylab("-log10(FDR)") + 
   ggtitle("MZ twins") + 
   theme_bw()
DAVID_cortex_GE_UP <- read.delim("~/快盘/FetalBrain/RNAseq/DEfine/gene/enrich/cortex_GE_UP_duplicated_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
DAVID_cortex_GE_DN <- read.delim("~/快盘/FetalBrain/RNAseq/DEfine/gene/enrich/cortex_GE_DN_duplicated_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
Terms <- unique(c(DAVID_cortex_GE_UP[DAVID_cortex_GE_UP$FDR < 0.01, "Term"], DAVID_cortex_GE_DN[DAVID_cortex_GE_DN$FDR < 0.01, "Term"]))
DAVID_neurospheres <- rbind(filter(DAVID_cortex_GE_UP, Term %in% Terms) %>% mutate(DE = "UP"), filter(DAVID_cortex_GE_DN, Term %in% Terms) %>% mutate(DE = "DN")) %>% mutate(Term = gsub("GO:\\d+~", "", Term), FDR = -log10(FDR) -2) %>% select(DE, Term, FDR)
DAVID_neurospheres$Term <- gsub("cell morphogenesis involved in neuron differentiation", "cell morphogenesis involved in\n neuron differentiation", DAVID_neurospheres$Term)
DAVID_neurospheres$Term <- gsub("cell morphogenesis involved in differentiation", "cell morphogenesis involved in\n differentiation", DAVID_neurospheres$Term)
DAVID_neurospheres$Term <- factor(DAVID_neurospheres$Term, levels = rev(unique(DAVID_neurospheres[order(DAVID_neurospheres$FDR, decreasing = T), "Term"])))
DAVID_neurospheres_figure <- ggplot(DAVID_neurospheres, aes(x = Term, y = FDR, fill = DE)) + 
   geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
   coord_flip() + 
   scale_fill_manual(values = c("blue", "red"), labels = c("UP in GE", "UP in Cortex"), name = "") + 
   scale_y_continuous(breaks = seq(-4, 6, by = 2), labels = seq(-2, 8, by = 2)) + 
   xlab("") + 
   ylab("-log10(FDR)") + 
   ggtitle("Neurospheres") + 
   theme_bw()
# write.table(c(GW_UP_UP_cortex$id, GW_UP_DN_cortex$id, GW_DN_UP_cortex$id, GW_DN_DN_cortex$id, GW_UP_no_cortex$id, GW_no_UP_cortex$id, GW_DN_no_cortex$id, GW_no_DN_cortex$id), file = "~/快盘/FetalBrain/GW/DE/GW_DE_cortex.txt", row.names = F, col.names = F, quote = F)
GW_DE_cortex_DAVID <- read.delim("~/快盘/FetalBrain/GW/DE/enrich/GW_DE_cortex_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
# write.table(c(GW_UP_UP_GE$id, GW_UP_DN_GE$id, GW_DN_UP_GE$id, GW_DN_DN_GE$id, GW_UP_no_GE$id, GW_no_UP_GE$id, GW_DN_no_GE$id, GW_no_DN_GE$id), file = "~/快盘/FetalBrain/GW/DE/GW_DE_GE.txt", row.names = F, col.names = F, quote = F)
GW_DE_GE_DAVID <- read.delim("~/快盘/FetalBrain/GW/DE/enrich/GW_DE_GE_DAVID.txt", as.is = T) %>% filter(Category == "GOTERM_BP_5")
Terms <- unique(c(GW_DE_cortex_DAVID[GW_DE_cortex_DAVID$FDR < 0.01, "Term"], GW_DE_GE_DAVID[GW_DE_GE_DAVID$FDR < 0.01, "Term"]))
DAVID_GW <- rbind(filter(GW_DE_cortex_DAVID, Term %in% Terms) %>% mutate(Cell = "Cortex"), filter(GW_DE_GE_DAVID, Term %in% Terms) %>% mutate(Cell = "GE")) %>% mutate(Term = gsub("GO:\\d+~", "", Term), FDR = -log10(FDR) -2) %>% select(Cell, Term, FDR)
DAVID_GW$Term <- gsub("cell morphogenesis involved in neuron differentiation", "cell morphogenesis involved in\n neuron differentiation", DAVID_GW$Term)
DAVID_GW$Term <- gsub("cell morphogenesis involved in differentiation", "cell morphogenesis involved in\n differentiation", DAVID_GW$Term)
DAVID_GW$Term <- factor(DAVID_GW$Term, levels = rev(unique(DAVID_GW[order(DAVID_GW$FDR, decreasing = T), "Term"])))
DAVID_GW_figure <- ggplot(DAVID_GW, aes(x = Term, y = FDR, fill = Cell)) + 
   geom_bar(stat = "identity", position = "dodge", width = 0.5) + 
   coord_flip() + 
   scale_fill_manual(values = c("red", "blue"), name = "") + 
   scale_y_continuous(breaks = seq(-4, 6, by = 2), labels = seq(-2, 8, by = 2)) + 
   xlab("") + 
   ylab("-log10(FDR)") + 
   ggtitle("GW") + 
   theme_bw()
grid.newpage()
layOut(list(DAVID_MZ_figure, 1:2, 1), 
       list(DAVID_neurospheres_figure, 1:4, 2), 
       list(DAVID_GW_figure, 3:5, 1))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.58, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS5.pdf", height = 12, width = 12)
layOut(list(DAVID_MZ_figure, 1:2, 1), 
       list(DAVID_neurospheres_figure, 1:4, 2), 
       list(DAVID_GW_figure, 3:5, 1))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.58, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 6: Validate WGBS UMRs with MeDIP/MRE

```{r WGBS_valid}
GOBP_Cortex02.UMR_MeDIP <- read.delim("~/快盘/FetalBrain/MeDIP/DMR/enrich/ALL_GREAT_GOBP_DMR.Cortex-HuFNSC02_GE-HuFNSC02.hypo.tsv", head = F, as.is = T, skip = 4)
GOBP_GE02.UMR_MeDIP <- read.delim("~/快盘/FetalBrain/MeDIP/DMR/enrich/ALL_GREAT_GOBP_DMR.Cortex-HuFNSC02_GE-HuFNSC02.hyper.tsv", head = F, as.is = T, skip = 4)
GOBP_Cortex02_GE02_MeDIP <- rbind(select(mutate(filter(GOBP_Cortex02.UMR_MeDIP, V8 >= 2 & V7 <= 0.05 & V16 <= 0.05), UMR = "Cortex.UMR", FDR = -log10(V7), Term = V3), UMR, Term, FDR),
                                 select(mutate(filter(GOBP_GE02.UMR_MeDIP, V8 >= 2 & V7 <= 0.05 & V16 <= 0.05), UMR = "GE.UMR", FDR = log10(V7), Term = V3), UMR, Term, FDR))
Terms <- unique(c(arrange(filter(GOBP_Cortex02_GE02_MeDIP, UMR == "Cortex.UMR"), desc(FDR))[1:20, "Term"], rev(arrange(filter(GOBP_Cortex02_GE02_MeDIP, UMR == "GE.UMR"), FDR)[1:20, "Term"])))
GOBP_Cortex02_GE02_MeDIP <- mutate(filter(GOBP_Cortex02_GE02_MeDIP, Term %in% Terms), Term = factor(Term, levels = rev(Terms)))
GOBP_Cortex02_GE02_MeDIP_figure <- ggplot(data = GOBP_Cortex02_GE02_MeDIP, aes(Term, FDR, fill = UMR)) +
  geom_bar(position = "identity", stat = "identity", width = .5) + 
  coord_flip() + 
  xlab("") + 
  ylab("-log10(FDR)") + 
  scale_fill_manual(values = c("blue", "red"), name = "") + 
  theme_bw()
grid.newpage()
layOut(list(valid_boxplot, 1, 1:2), 
       list(DMR_freq_neurospheres_figure + scale_fill_manual(values = c("blue", "red"), labels = c("GE UMRs", "Cortex UMRs"), name = ""), 1, 3:4), 
       list(GOBP_Cortex02_GE02_MeDIP_figure, 2, 1:3))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS6.pdf", height = 12, width = 12)
layOut(list(valid_boxplot, 1, 1:2), 
       list(DMR_freq_neurospheres_figure + scale_fill_manual(values = c("blue", "red"), labels = c("GE UMRs", "Cortex UMRs"), name = ""), 1, 3:4), 
       list(GOBP_Cortex02_GE02_MeDIP_figure, 2, 1:3))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 7: Location of UMRs across the genome 

```{r UMR_pos}
DMR_pos_MZ_ggplot <- DMR_pos_MZ_figure@ggplot + 
  theme(axis.title.x = element_text(size = 10), axis.text.x = element_text(angle = 90, size = 8), strip.text.x = element_text(size = 8), strip.text.y = element_text(angle = 0, size = 8), plot.title = element_text(size = 10), legend.title = element_text(size = 0), legend.text = element_text(size = 8), panel.margin = unit(0.1, "lines")) + 
  ggtitle("UMRs between MZ twins") + 
  xlab("Position of UMRs on the chromosome") + 
  scale_color_manual(values = c("red", "blue"), labels = c("HuFNSC02 UMRs", "HuFNSC01 UMRs"), name = "") + 
  guides(color = guide_legend(override.aes = list(size=5, alpha = 1)), fill = 'none')
DMR_pos_neurospheres_ggplot <- DMR_pos_WGBS_figure@ggplot + 
  theme(axis.title.x = element_text(size = 10), axis.text.x = element_text(angle = 90, size = 8), strip.text.x = element_text(size = 8), strip.text.y = element_text(angle = 0, size = 8), plot.title = element_text(size = 10), legend.title = element_text(size = 0), legend.text = element_text(size = 8), panel.margin = unit(0.1, "lines")) + 
  ggtitle("UMRs between neurospheres") + 
  xlab("Position of UMRs on the chromosome") + 
  scale_color_manual(values = c("blue", "red"), labels = c("GE UMRs", "Cortex UMRs"), name = "") + 
  guides(color = guide_legend(override.aes = list(size=5, alpha = 1)), fill = 'none')
DMR_pos_GW_ggplot <- DMR_pos_GW_figure@ggplot + 
  theme(axis.title.x = element_text(size = 10), axis.text.x = element_text(angle = 90, size = 8), strip.text.x = element_text(size = 8), strip.text.y = element_text(angle = 0, size = 8), plot.title = element_text(size = 10), legend.title = element_text(size = 0), legend.text = element_text(size = 8), panel.margin = unit(0.1, "lines")) + 
  ggtitle("UMRs between GW") + 
  xlab("Position of UMRs on the chromosome") + 
  scale_color_manual(values = c("red", "blue"), labels = c("GW13 UMRs", "GW17 UMRs"), name = "") + 
  guides(color = guide_legend(override.aes = list(size=5, alpha = 1)), fill = 'none')
grid.newpage()
layOut(list(DMR_pos_MZ_ggplot, 1, 1:3), 
       list(DMR_pos_neurospheres_ggplot, 2, 1:2), 
       list(DMR_pos_GW_ggplot, 2, 3:4))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.515, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS7.pdf", height = 12, width = 12)
layOut(list(DMR_pos_MZ_ggplot, 1, 1:3), 
       list(DMR_pos_neurospheres_ggplot, 2, 1:2), 
       list(DMR_pos_GW_ggplot, 2, 3:4))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.515, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 8: Neurosphere UMR enrichemnt at chromosome ends

```{r chrEnd_WGBS, fig.height = 6, fig.width = 12}
chrEnd_Cortex02UMR <- read.delim("~/快盘/FetalBrain/WGBS/DMR/DMR.Cortex-HuFNSC02_GE-HuFNSC02.m0.75.p0.005.d0.5.s300.c3.hypo.bed.chrom.window", head = F, as.is = T)
chrEnd_GE02UMR <- read.delim("~/快盘/FetalBrain/WGBS/DMR/DMR.Cortex-HuFNSC02_GE-HuFNSC02.m0.75.p0.005.d0.5.s300.c3.hyper.bed.chrom.window", head = F, as.is = T)
chrEnd_Cortex04UMR <- read.delim("~/快盘/FetalBrain/WGBS/DMR/DMR.Cortex-HuFNSC04_GE-HuFNSC04.m0.75.p0.005.d0.5.s300.c3.hypo.bed.chrom.window", head = F, as.is = T)
chrEnd_GE04UMR <- read.delim("~/快盘/FetalBrain/WGBS/DMR/DMR.Cortex-HuFNSC04_GE-HuFNSC04.m0.75.p0.005.d0.5.s300.c3.hyper.bed.chrom.window", head = F, as.is = T)
chrEnd <- rbind(data.frame(DM = "Cortex.UMRs", Donor = "HuFNSC02", chrEnd_Cortex02UMR %>% group_by(V4) %>% summarise(N = sum(V5))), data.frame(DM = "GE.UMRs", Donor = "HuFNSC02", chrEnd_GE02UMR %>% group_by(V4) %>% summarise(N = sum(V5))), data.frame(DM = "Cortex.UMRs", Donor = "HuFNSC04", chrEnd_Cortex04UMR %>% group_by(V4) %>% summarise(N = sum(V5))), data.frame(DM = "GE.UMRs", Donor = "HuFNSC04", chrEnd_Cortex04UMR %>% group_by(V4) %>% summarise(N = sum(V5))))
chrEnd_WGBS_figure <- ggplot(chrEnd, aes(x = Donor, y = N, fill = as.factor(V4))) + 
   geom_bar(stat = "identity", width = 0.5, position = "fill") + 
   facet_grid(DM ~ .) + 
   xlab("") + 
   ylab("Fraction of total UMRs") + 
   ggtitle("Neurosphere UMRs") + 
   coord_flip() + 
   scale_fill_discrete(name = "Normalized length \nalong chromosome", labels = as.character(seq(0.1, 1, by = 0.1)), guide = F) + 
   theme_bw() + 
   theme(strip.text = element_text(size = 15), axis.title.y = element_text(size = 15), axis.text = element_text(size = 12), legend.text = element_text(size = 15))
chrEnd_GW_figure <- ggplot(chrEnd_GW_figure$data %>% mutate(DM = revalue(DM, c("17-week.UMRs" = "GW17 UMRs", "13-week.UMRs" = "GW13 UMRs"))), aes(x = Cell, y = N, fill = as.factor(V4))) + 
   geom_bar(stat = "identity", width = 0.5, position = "fill") + 
   facet_grid(DM ~ .) + 
   xlab("") + 
   ylab("Fraction of total UMRs") + 
   ggtitle("GW UMRs") + 
   coord_flip() + 
   scale_fill_discrete(name = "Normalized length \nalong chromosome", labels = as.character(seq(0.1, 1, by = 0.1))) + 
   theme_bw() + 
   theme(strip.text = element_text(size = 15), axis.title.y = element_text(size = 15), axis.text = element_text(size = 12), legend.text = element_text(size = 15))
grid.arrange(chrEnd_WGBS_figure, chrEnd_GW_figure, nrow = 1, widths = 5:6)
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.48, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS8.pdf", height = 6, width = 12)
grid.arrange(chrEnd_WGBS_figure, chrEnd_GW_figure, nrow = 1, widths = 5:6)
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.48, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 9: DAVID enrichment for isoform genes

```{r isoform_DAVID}
isoform_neurospheres_DAVID <- isoform_cortex_GE_enrich$figure + ggtitle("Neurospheres") + xlab("") + 
  theme(plot.margin=unit(c(1,1.35,1,1), "cm"))
enrich_GW_isoform_gene_dup <- enrich_GW_isoform_gene_dup + ggtitle("GW") + xlab("") + 
  theme(plot.margin=unit(c(1,1,1,2.5), "cm"))
grid.arrange(isoform_neurospheres_DAVID, enrich_GW_isoform_gene_dup, nrow = 2, heights = 9:5)
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.42, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS9.pdf", height = 12, width = 12)
grid.arrange(isoform_neurospheres_DAVID, enrich_GW_isoform_gene_dup, nrow = 2, heights = 9:5)
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.42, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

##################################################################

### Supplemental 10: TFBS asymmetry in MZ UMRs

```{r MZ_TFBS}
DMR_MZ_TF_figure <- DMR_MZ_TF_figure + xlab("Abundance near HuFNSC02 UMRs") + ylab("Abundance near HuFNSC01 UMRs")
DMR_GW_TF_figure <- DMR_GW_TF_figure + xlab("Abundance near GW17 UMRs") + ylab("Abundance near GW13 UMRs") + scale_color_manual(values = c("blue", "green"), labels = c("GW17 enriched", "Equal"))
grid.newpage()
layOut(list(DMR_MZ_TF_figure, 1, 1:4), 
       list(DMR_GW_TF_figure, 2, 1:3))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS10.pdf", height = 12, width = 12)
layOut(list(DMR_MZ_TF_figure, 1, 1:4), 
       list(DMR_GW_TF_figure, 2, 1:3))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```

### Supplemental 11: DNA methylation at exon boundaries

```{r neurospheres_epiProfile_5mC}
gt <- ggplot_gtable(ggplot_build(WGBS_figure_HuFNSC02 + ggtitle("WGBS HuFNSC02 cortex vs GE"))) 
gt$heights[[4]] <- unit(3.5, "null") 
grid.newpage()
grid.draw(gt) 
grid.text("Average DNA methylation", x = unit(0.015, "npc"), y = unit(0.65, "npc"), rot = 90)
grid.text("CpG density", x = unit(0.015, "npc"), y = unit(0.15, "npc"), rot = 90)
pdf("~/快盘/FetalBrain/Figures/FigureS11.pdf", height = 12, width = 12)
grid.draw(gt) 
grid.text("Average DNA methylation", x = unit(0.015, "npc"), y = unit(0.65, "npc"), rot = 90)
grid.text("CpG density", x = unit(0.015, "npc"), y = unit(0.15, "npc"), rot = 90)
dev.off()
```

### Supplemental 12: DNA methylation at exon boundaries with MeDIP

```{r neurospheres_epiProfile_5mC_MeDIP, fig.height = 6}
MeDIP_figure_HuFNSC01 <- MeDIP_figure_HuFNSC01 + ggtitle("HuFNSC01") + theme(legend.position = "none")
MeDIP_figure_HuFNSC02 <- MeDIP_figure_HuFNSC02 + ggtitle("HuFNSC02")
grid.arrange(MeDIP_figure_HuFNSC01, MeDIP_figure_HuFNSC02, nrow = 1, widths = 2:3)
pdf("~/快盘/FetalBrain/Figures/FigureS12.pdf", height = 6, width = 12)
grid.arrange(MeDIP_figure_HuFNSC01, MeDIP_figure_HuFNSC02, nrow = 1, widths = 2:3)
dev.off()
```

### Supplemental 13: Expression of DNA methylation reulators

```{r heatmap_5mC_regulator}
regulator_5mC <- read.delim("~/快盘/FetalBrain/RNAseq/regulator_5mC.rpkm", as.is = T, row.names = 1)
regulator_5mC <- as.matrix(regulator_5mC[, 2:ncol(regulator_5mC)])
(regulator_5mC_heatmap <- ggplot(melt(t(scale(t(regulator_5mC), center = T, scale = T))) %>% mutate(Gene = factor(X1, levels = rev(rownames(regulator_5mC))), Sample = factor(X2, levels = colnames(regulator_5mC))), aes(x=Sample, y=Gene, fill=value)) + 
   geom_tile() + 
   scale_fill_gradient(name = "Z-score") + 
   xlab("") + 
   ylab("") + 
   theme_bw() + 
   theme(axis.ticks.y = element_line(size = 0), panel.border = element_rect(size = 0), axis.text.x = element_text(angle = 90)))
ggsave(regulator_5mC_heatmap, file = "~/快盘/FetalBrain/Figures/FigureS13.pdf", height = 12, width = 12)
```

### Supplemental 14: RNA-seq and miRNA-seq clustering

```{r cluster}
gene_FetalBrain <- read.delim("~/快盘/FetalBrain/Tables/rpkm_pc.txt", as.is = T, row.names = 1)
gene_dend <- (1- (gene_FetalBrain %>% as.matrix %>% cor(method = "spearman"))) %>% as.dist %>% hclust %>% as.dendrogram %>% 
  set("by_labels_branches_col", value = c("Brain.HuFNSC01", "Brain.HuFNSC02"), TF_value = "green", type = "all") %>% 
  set("by_labels_branches_col", value = c("Cortex.HuFNSC01", "Cortex.HuFNSC02", "Cortex.HuFNSC03", "Cortex.HuFNSC04"), TF_value = "red", type = "all") %>% 
  set("by_labels_branches_col", value = c("GE.HuFNSC01", "GE.HuFNSC02", "GE.HuFNSC03", "GE.HuFNSC04"), TF_value = "blue", type = "all") %>% 
  set("by_labels_branches_lwd", value = colnames(gene_FetalBrain), TF_value = 2)
exon_FetalBrain <- read.delim("~/快盘/FetalBrain/Tables/rpkm_exon.txt", as.is = T, row.names = 1)
exon_dend <- (1- (exon_FetalBrain %>% as.matrix %>% cor(method = "spearman"))) %>% as.dist %>% hclust %>% as.dendrogram %>% 
  set("by_labels_branches_col", value = c("Brain.HuFNSC01", "Brain.HuFNSC02"), TF_value = "green", type = "all") %>% 
  set("by_labels_branches_col", value = c("Cortex.HuFNSC01", "Cortex.HuFNSC02", "Cortex.HuFNSC03", "Cortex.HuFNSC04"), TF_value = "red", type = "all") %>% 
  set("by_labels_branches_col", value = c("GE.HuFNSC01", "GE.HuFNSC02", "GE.HuFNSC03", "GE.HuFNSC04"), TF_value = "blue", type = "all") %>% 
  set("by_labels_branches_lwd", value = colnames(exon_FetalBrain), TF_value = 2)
op <- par(mar = c(5, 4, 4, 8), mfrow = c(2,2))
plot(gene_dend, main = "RNA-seq\nGene", horiz = TRUE)
mtext("a", side = 3, line = 2.5, cex = 1.5, adj = 0, padj = 0)
plot(exon_dend, main = "RNA-seq\nExon", horiz = TRUE)
mtext("b", side = 3, line = 2.5, cex = 1.5, adj = 0, padj = 0)
plot(miR_dend, main = "miRNA-seq\n(RPM > 100)", horiz = TRUE)
mtext("c", side = 3, line = 2.5, cex = 1.5, adj = 0, padj = 0)
plot.new()
legend("topleft", c("Brain", "Cortex", "GE"), col = c("green", "red", "blue"), lwd = 8)
par(op)
pdf("~/快盘/FetalBrain/Figures/FigureS14.pdf", height = 12, width = 12)
par(mar = c(5, 4, 4, 8), mfrow = c(2,2))
plot(gene_dend, main = "RNA-seq\nGene", horiz = TRUE)
mtext("a", side = 3, line = 2.5, cex = 1.5, adj = 0, padj = 0)
plot(exon_dend, main = "RNA-seq\nExon", horiz = TRUE)
mtext("b", side = 3, line = 2.5, cex = 1.5, adj = 0, padj = 0)
plot(miR_dend, main = "miRNA-seq\n(RPM > 100)", horiz = TRUE)
mtext("c", side = 3, line = 2.5, cex = 1.5, adj = 0, padj = 0)
plot.new()
legend("topleft", c("Brain", "Cortex", "GE"), col = c("green", "red", "blue"), lwd = 8)
dev.off()
```

### Supplemental 15: miRNA 

```{r miR}
miR_summary_figure <- miR_summary_figure + ylab("No. of miRs") + 
  theme(axis.text.x = element_text(angle = 90))
miR_DE_MZ_heatmap <- ggplot(melt(t(scale(t(miR_DE_MZ_heat), center = T, scale = T))) %>% mutate(miR = factor(X1, levels = rev(rownames(miR_DE_MZ_heat))), Sample = factor(X2, levels = colnames(miR_DE_MZ_heat))), aes(x=Sample, y=miR, fill=value)) + 
   geom_tile() + 
   scale_fill_gradient(name = "Z-score") + 
   xlab("") + 
   ylab("") + 
   ggtitle("MZ twins") + 
   theme_bw() + 
   theme(axis.text.y = element_text(size = 6), axis.ticks.y = element_line(size = 0), panel.border = element_rect(size = 0), axis.text.x = element_text(angle = 90))
miR_DE_neurospheres_heatmap <- ggplot(melt(t(scale(t(miR_DE_neurospheres_heat), center = T, scale = T))) %>% mutate(miR = factor(X1, levels = rev(rownames(miR_DE_neurospheres_heat))), Sample = factor(X2, levels = colnames(miR_DE_neurospheres_heat))), aes(x=Sample, y=miR, fill=value)) + 
   geom_tile() + 
   scale_fill_gradient(name = "Z-score") + 
   xlab("") + 
   ylab("") + 
   ggtitle("Neurospheres") + 
   theme_bw() + 
   theme(axis.text.y = element_text(size = 6), axis.ticks.y = element_line(size = 0), panel.border = element_rect(size = 0), axis.text.x = element_text(angle = 90))
miR_DE_GW_heatmap <- ggplot(melt(t(scale(t(miR_DE_GW_heat), center = T, scale = T))) %>% mutate(miR = factor(X1, levels = rev(rownames(miR_DE_GW_heat))), Sample = factor(X2, levels = colnames(miR_DE_GW_heat))), aes(x=Sample, y=miR, fill=value)) + 
   geom_tile() + 
   scale_fill_gradient(name = "Z-score") + 
   xlab("") + 
   ylab("") + 
   ggtitle("GW") + 
   theme_bw() + 
   theme(axis.text.y = element_text(size = 5), axis.ticks.y = element_line(size = 0), panel.border = element_rect(size = 0), axis.text.x = element_text(angle = 90))
grid.arrange(miR_summary_figure, miR_DE_MZ_heatmap, miR_DE_neurospheres_heatmap, miR_DE_GW_heatmap, nrow = 2)
grid.text("a", x=unit(.02, "npc"), y=unit(.98, "npc"), gp=gpar(fontsize=20))
grid.text("b", x=unit(.52, "npc"), y=unit(.98, "npc"), gp=gpar(fontsize=20))
grid.text("c", x=unit(.02, "npc"), y=unit(.48, "npc"), gp=gpar(fontsize=20))
grid.text("d", x=unit(.52, "npc"), y=unit(.48, "npc"), gp=gpar(fontsize=20))
pdf("~/快盘/FetalBrain/Figures/FigureS15.pdf", height = 12, width = 12)
grid.arrange(miR_summary_figure, miR_DE_MZ_heatmap, miR_DE_neurospheres_heatmap, miR_DE_GW_heatmap, nrow = 2)
grid.text("a", x=unit(.02, "npc"), y=unit(.98, "npc"), gp=gpar(fontsize=20))
grid.text("b", x=unit(.52, "npc"), y=unit(.98, "npc"), gp=gpar(fontsize=20))
grid.text("c", x=unit(.02, "npc"), y=unit(.48, "npc"), gp=gpar(fontsize=20))
grid.text("d", x=unit(.52, "npc"), y=unit(.48, "npc"), gp=gpar(fontsize=20))
dev.off()
```

### Supplemental 16: GREAT for GW UMRs  

```{r GREAT_GW}
GREAT_GW_Cortex <- rbind(GREAT_GW_Cortex02.UMR$data %>% filter(Category == "GOBP") %>% mutate(UMR = "GW17 UMRs", FDR = -log(FDR)), 
                         GREAT_GW_Cortex04.UMR$data %>% filter(Category == "GOBP") %>% mutate(UMR = "GW13 UMRs", FDR = log(FDR))) %>% arrange(-FDR) %>% mutate(Term = factor(Term, levels = rev(Term)))
GREAT_GW_Cortex_figure <- ggplot(data = GREAT_GW_Cortex, aes(Term, FDR, fill = UMR)) +
  geom_bar(position = "identity", stat = "identity", width = .5) + 
  coord_flip() + 
  xlab("") + 
  ylab("-log10(FDR)") + 
  ggtitle("Cortex") + 
  scale_fill_manual(values = c("red", "blue"), name = "", guide = F) + 
  theme_bw()
GREAT_GW_GE <- rbind(GREAT_GW_GE02.UMR$data %>% filter(Category == "GOBP") %>% mutate(UMR = "GW17 UMRs", FDR = -log(FDR)), 
                     GREAT_GW_GE04.UMR$data %>% filter(Category == "GOBP") %>% mutate(UMR = "GW13 UMRs", FDR = log(FDR))) %>% arrange(-FDR) %>% mutate(Term = factor(Term, levels = rev(Term)))
GREAT_GW_GE_figure <- ggplot(data = GREAT_GW_GE, aes(Term, FDR, fill = UMR)) +
  geom_bar(position = "identity", stat = "identity", width = .5) + 
  coord_flip() + 
  xlab("") + 
  ylab("-log10(FDR)") + 
  ggtitle("GE") + 
  scale_fill_manual(values = c("red", "blue"), name = "") + 
  scale_y_continuous(breaks = c(-20, 0, 20), labels = c(20, 0, 20)) + 
  theme_bw()
GREAT_GW_shared <- GREAT_GW_17week.UMR$data %>% filter(Category == "GOBP") %>% mutate(FDR = -log(FDR)) %>% arrange(-FDR) %>% mutate(Term = factor(Term, levels = rev(Term)))
GREAT_GW_shared_figure <- ggplot(data = GREAT_GW_shared, aes(Term, FDR)) +
  geom_bar(position = "identity", stat = "identity", width = .5, fill = "blue") + 
  coord_flip() + 
  xlab("") + 
  ylab("-log10(FDR)") + 
  ggtitle("shared") + 
  theme_bw()
grid.newpage()
layOut(list(GREAT_GW_Cortex_figure, 1:2, 1:4), 
       list(GREAT_GW_GE_figure, 1:3, 5:9), 
       list(GREAT_GW_shared_figure, 3:4, 1:4))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
pdf("~/快盘/FetalBrain/Figures/FigureS16.pdf", height = 12, width = 12)
layOut(list(GREAT_GW_Cortex_figure, 1:2, 1:4), 
       list(GREAT_GW_GE_figure, 1:3, 5:9), 
       list(GREAT_GW_shared_figure, 3:4, 1:4))
grid.text("a", x = unit(0.015, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("b", x = unit(0.515, "npc"), y = unit(0.98, "npc"), gp=gpar(fontsize = 20))
grid.text("c", x = unit(0.015, "npc"), y = unit(0.48, "npc"), gp=gpar(fontsize = 20))
dev.off()
```


<!--
### Supplemental : GREAT analysis on UMRs between MZ twins - GOBP 

```{r MZ_GREAT, eval = F}
GREAT_MZ <- rbind(data.frame(UMR = "HuFNSC01.UMRs", Cell = "Brain", GREAT_HuFNSC01.UMR.Brain$data), data.frame(UMR = "HuFNSC02.UMRs", Cell = "Brain", GREAT_HuFNSC02.UMR.Brain$data), data.frame(UMR = "HuFNSC01.UMRs", Cell = "Cortex", GREAT_HuFNSC01.UMR.Cortex$data), data.frame(UMR = "HuFNSC02.UMRs", Cell = "Cortex", GREAT_HuFNSC02.UMR.Cortex$data), data.frame(UMR = "HuFNSC01.UMRs", Cell = "GE", GREAT_HuFNSC01.UMR.GE$data), data.frame(UMR = "HuFNSC02.UMRs", Cell = "GE", GREAT_HuFNSC02.UMR.GE$data))
GREAT_MZ <- filter(GREAT_MZ, Category == "GOBP")
GREAT_MZ$FDR <- -log10(GREAT_MZ$FDR)
GREAT_MZ[GREAT_MZ$UMR == "HuFNSC01.UMRs", "FDR"] <- -GREAT_MZ[GREAT_MZ$UMR == "HuFNSC01.UMRs", "FDR"]
(GREAT_MZ_figure <- ggplot(data = GREAT_MZ, aes(Term, FDR)) +
  geom_bar(aes(fill = UMR), stat = "identity", width = .5, position = "identity") + 
  coord_flip() + 
  facet_grid(Cell ~ ., scales = "free_x") + 
  xlab("") + 
  theme_bw())
ggsave(GREAT_MZ_figure, file = "~/快盘/FetalBrain/Figures/FigureS.pdf", height = height, width = width)
```
-->




