---
title: "CB65_MN1ND13 DP vs DN - DNA methylation"
author: "Gloria Li"
date: "Nov 30, 2016"
output: 
  html_document:
    keep_md: true
    toc: true
    toc_float: true
    toc_depth: 4
---

Updated: `r date()`

```{r setup, include = FALSE}
library(ggplot2)
library(plyr)
library(VennDiagram)
library(gridExtra)
library(gplots)
library(dendextend)
library(reshape)
library(wq)
library(dplyr)
library(RCircos)
library(knitr)
load("/projects/epigenomics2/PING/DMR/DMR.Rdata")
knitr::opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE, results = FALSE, fig.height = 5, fig.width = 5) 
knitr::opts_knit$set(root.dir = "/projects/epigenomics2/PING/DMR/")
```

## CpG coverage QC 
* DN library has lower coverage than the others at CpGs.  
* No. of CpGs with coverage $\ge$ 3:        
	+ DP:   26429266
	+ DN:   24902572
	+ CD34: 27017534

```{r coverage}
(CpG_coverage_figure)
```

## Global methylation level
* Both DP and DN have higher methylation level than CD34+ cells globally.      
* DP has higher methylation level than DN globally.      

```{r global}
(quantile_5mC_figure)
(delta_5mC_figure)
```

## DMRs 
### Method
* Identify DM CpGs     
	+ methyl_diff p-value < 0.005  
	+ delta fractional methylation > 0.5  
	+ fractional methylation of one sample > 0.6   
* Collapse DM CpGs into DMRs     
	+ adjacent DM CpGs have the same DM status;    
	+ distance between adjacent CpGs (size) < 500bp;   
	+ No. of CpGs within each DMR $\ge$ 3.   

### DMR QC 
* All DMR QC metrics seem within normal range.    
* DMRs between DP and DN are of shorter length, but it can be expected due to the fewer number of DM CpGs.       

```{r DMR_QC, fig.width = 7}
(DMR_DP_DN_figure$length)
(DMR_DP_CD34_figure$length)
(DMR_DN_CD34_figure$length)
```

### Summary
* No. of DMRs:

Samples | Hypermethylated | Hypomethylated
------- | --------------- | --------------
DP vs DN | 2317 | 12
DP vs CD34+ | 18028 | 2971
DN vs CD34+ | 14864 | 5009

```{r DMR_summary}
(DMR_DP_DN_figure$freq)
(DMR_DP_CD34_figure$freq)
(DMR_DN_CD34_figure$freq)
```

### Enrichment in genomic locations
* DMRs between DP and DN:       
	+ DMRs hypermethylated in DP do not seem to be enriched for any specific genomic locations, but seem to be depleted from CGIs.     
	+ Enrichment for DMRs hypomethylated in DP is not reliable due to small sample size.     
* DMRs between DP/DN and CD34+:       
	+ Both hyper and hypo DMRs are enriched in CGIs, CGI shores, promoters, and exons.         

```{r DMR_genomicBreak}
(genomic_breakdown_figure)
```

### GREAT functional enrichment analysis
* DMRs between DP and DN:       
	+ DMRs hypermethylated in DP are enriched in immune response.     
* DMRs between DP/DN and CD34+:     
	+ DMRs hypermethylated in DP/DN are enriched in immune system development and cell differentiation.       
	+ DMRs hypomethylated in DP are enriched in cell signaling.      
	+ DMRs hypomethylated in DN are enriched in immunity.    

```{r DMR_GREAT, fig.height=10, fig.width=8}
(GREAT_DMR_DP_DN_hyper_figure)
(GREAT_DMR_DP_CD34_hyper_figure)
(GREAT_DMR_DP_CD34_hypo_figure)
(GREAT_DMR_DN_CD34_hyper_figure)
(GREAT_DMR_DN_CD34_hypo_figure)
```

