---
title: "Glioma - Histone modifications ChIP-seq"
author: "Gloria Li"
date: "Jun, 19, 2016"
output: 
  html_document:
    keep_md: true
    toc: true
    toc_float: true
    toc_depth: 4
    code_folding: hide
---

Updated: `r date()`

```{r setup, include = FALSE}
library(ggplot2)
library(plyr)
library(VennDiagram)
library(gridExtra)
library(gplots)
library(dendextend)
library(reshape)
library(wq)
library(dplyr)
library(RCircos)
library(knitr)
load("/projects/epigenomics2/users/lli/glioma/ChIPseq/ChIPseq.Rdata")
knitr::opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE, results = FALSE, fig.height = 8, fig.width = 7, bitmapType='cairo') 
knitr::opts_knit$set(root.dir = "/projects/epigenomics2/users/lli/glioma/ChIPseq/")
libs <- c("CEMT_19", "CEMT_21", "CEMT_22", "CEMT_23", "CEMT_47")
```

## Global comparison
* Calculate an average signal of 500bp bins for both samples.           
* Look into only bins that that have non zero signal in both.       
* Find bins that are common in top 400 bins in both samples, likely blacklist regions, e.g. tandem repeats.       
* Plot box plot for the ratio log2(signal2/signal1) in top bins.       
* Compare it the same ratio for the rest of the bins.       
* Results and conclusion: inconsistency in input samples suggest that this method is not accurate when comparing samples from different biological origins, maybe due to genetic differences in these blacklist regions, e.g. different number of copies of the repeats. Thus these top ranked regions cannot be used as normalization standard.   

```{r global}
(global_H3K36me3_figure)
(global_H3K27me3_figure)
(global_Input_figure)
```

## FindER enriched regions 
* FindER v 1.0.0b      

```{r ER_summary}
(ER_summary_figure)
```

## Unique enriched regions 
* Pairwise comparisons between glioma samples (S1) and NPC GE04 (S2).   
* Challenge: S1 and S2 often have quite different sequencing depth, thus direct comparisons between ER are likely biased.     
* Input: ER in S1, wig in S1, ER in S2, wig in S2.   
* Method: 
	+ Calculate S1 ER signal in S2 wig.    
	+ Randomly generate a set of background regions (not overlapping with S2 ER) with same length as S1 ER.   
	+ Calculate signal for these background regions in S2 wig, use 90% quantile as cutoff. 
	+ S1 unique ER: S1 ER with signal < background cutoff in S2.   
	
```{r unique_ER_summary, fig.height=5}
(ER_unique_summary_figure)
```

### Loss of H3K36me3      
* Global loss of H3K36me3 in both IDH mutant and wt gliomas.   
* H3K36me3 loss regions are enriched in genebody but not intergenic regions, while H3K36me3 gain regions are also enriched in intergenic regions.      

```{r K36_loss}
(H3K36me3_unique_summary_figure)
(H3K36me3_unique_enrich_summary_figure)
```

#### H3K36me3 loss regions are consistent in all glioma samples

```{r K36_intersect, results='asis'}
H3K36me3_loss_intersect_venn <- draw.pairwise.venn(area1 = H3K36me3_loss_intersect_summary[1, "N1"], area2 = H3K36me3_loss_intersect_summary[1, "N2"], cross.area = H3K36me3_loss_intersect_summary[1, "Intersect"], category = c(H3K36me3_loss_intersect_summary[1, "Sample1"], H3K36me3_loss_intersect_summary[1, "Sample2"]), fill = c("red", "blue"))
(H3K36me3_loss_intersect_figure)
kable(H3K36me3_loss_intersect_summary %>% select(-samples), digits = 4, format = "html", align = "c", row.names = F)
```

#### H3K36me3 loss regions are associated with cell cycle regulation

```{r K36_loss_GREAT}
(H3K36me3_unique_GREAT)
(H3K36me3_loss_gene_DAVID)
```

#### H3K36me3 loss regions are associated with transcriptional downregulation

```{r K36_loss_RPKM, results='asis'}
(H3K36me3_unique_NPC_gene_RPKM_figure)
kable(H3K36me3_loss_DE_summary, digits = 4, format = "html", align = "c", row.names = F)
```

#### Subtle gain of H3K27me3   
* No significant correlation between H3K36me3 and H3K27me3 signals.    
* The intersect of H3K27me3 gain/loss regions between different samples are still significant.     
* Both H3K27me3 gain and loss regions are enriched in genebody.       
* The functions of genes associated with H3K27me3 gain/loss regions are different.      

```{r unique_ER_K36_K27, results='asis'}
(H3K36me3_unique_NPC_K27me3_K36me3_CEMT_19_density2d)
kable(H3K36me3_loss_H3K27me3_intersect_summary, digits = 4, format = "html", align = "c", row.names = F)
(H3K36me3_loss_H3K27me3_summary_figure)
(H3K36me3_loss_H3K27me3gain_GREAT)
(H3K36me3_loss_H3K27me3loss_GREAT)
```

#### No redistribution of H3K27me3      

```{r K27me3_redistribution}
(HM_distrbution_figure)
```

* [Nada's results](http://science.sciencemag.org/content/352/6287/844.figures-only)   

### Histone modification associated DE genes
* As expected, active marks, i.e. H3K27ac and H3K4me3 were significantly associated with transcriptional activation, while H3K27me3 was associated with down regulation.    

```{r unique_ER_DE}
(DHM_DE_figure)
```

### Unique enhancers and transcription factor activities
* Homer transcription factor binding enrichment analysis for unique enhancers.   
	+ q-value < 0.01
	+ Fraction of enhancers with motif > 20%

#### H3K27ac
* Four transcription factors enriched in NPC-specific H3K27ac, only Sox3 was expressed in NPCs.     
	+ Sox3: function as a switch in neuronal development. Keeps neural cells undifferentiated by counteracting the activity of proneural proteins and suppresses neuronal differentiation.      

```{r unique_H3K27ac_NPC, fig.height=4, results='asis'}
(homer_unique_K27ac_NPC_tf_figure)
(homer_unique_K27ac_NPC_tf_RPKM_figure)
kable(homer_unique_K27ac_NPC_Sox3_DN %>% select(ENSG, Name), digits = 4, format = "html", align = "l", row.names = F)
```
```{r unique_H3K27ac_NPC_DAVID}
(homer_unique_K27ac_NPC_Sox3_5mC_heatmap + ggtitle("Sox3 enhancer fractional methylation"))
(homer_unique_K27ac_NPC_Sox3_DN_DAVID)
```

* No significantly enriched transcription factor in glioma-specific H3K27ac.   

#### H3K4me1
* Three transcription factors were enirched and expressed in NPC-specific H3K4me1: Sox3, Sox6, Lhx2.     
	+ Sox3: function as a switch in neuronal development. Keeps neural cells undifferentiated by counteracting the activity of proneural proteins and suppresses neuronal differentiation.      
	+ Sox6: plays a key role in several developmental processes, including neurogenesis and skeleton formation.      
	+ Lhx2: acts as a transcriptional activator. Transcriptional regulatory protein involved in the control of cell differentiation in developing lymphoid and neural cell types.        
	
```{r unique_H3K4me1_NPC}
(homer_unique_K4me1_NPC_tf_figure)
(homer_unique_K4me1_NPC_tf_RPKM_figure)
```
```{r unique_H3K4me1_NPC_DAVID, fig.height=2}
(homer_unique_K4me1_NPC_Sox3_DN_DAVID)
(homer_unique_K4me1_NPC_Sox6_DN_DAVID)
(homer_unique_K4me1_NPC_Lhx2_DN_DAVID)
```

* Three transcription factors were enriched and expressed in glioma-specific H3K4me1: HEB (TCF12), Ascl1, Olig2.      
	+ HEB: involved in the initiation of neuronal differentiation.     
	+ Ascl1: plays a role in the neuronal commitment and differentiation and in the generation of olfactory and autonomic neurons.      
	+ Olig2: is an essential regulator of ventral neuroectodermal progenitor cell fate. Required for oligodendrocyte and motor neuron specification in the spinal cord, as well as for the development of somatic motor neurons in the hindbrain.      
```{r unique_H3K4me1_IDHmut}
(homer_unique_K4me1_IDHmut_tf_figure)
(homer_unique_K4me1_IDHmut_tf_RPKM_figure)
(homer_unique_K4me1_IDHmut_Ascl1_UP_DAVID)
(homer_unique_K4me1_IDHmut_Olig2_UP_DAVID)
(homer_unique_K4me1_IDHmut_HEB_UP_DAVID)
```

## Chromatin states
* ChromHMM for 6 core histone marks in glioma and GE04.    
![](/projects/epigenomics2/users/lli/glioma/ChIPseq/ChromHMM/emission.pdf)

```{r chromHMM}
(chromHMM_summary_figure)
```

