---
output: 
  html_document:
    keep_md: true
    toc: yes
---
Fetal Brain isoform analysis - epiProfile
========================================================

Gloria Li         
`r date()` 

<!-- re-knit after modify epiProfile.R script -->

```{r include = FALSE}
library(knitr)
opts_chunk$set(message=FALSE, echo = FALSE, warning = FALSE)
```

```{r}
setwd("~/快盘/FetalBrain/RNAseq/epiProfile/")
source("~/HirstLab/Pipeline/epiProfile.R")
library(ggplot2)
library(grid) 
load("exonProfile_Rmd.Rdata")
```

## Between cortex and GE neurospheres
### DNA methylation at exon boundaries 
* 5mC at cassette exon boundaries has similar pattern to expressed in both exons.   
* 5mC for cassette exons between cortex and GE neurospheres shows no significant differences.   
* Results from both WGBS (HuFNSC02 & HuFNSC04) and MeDIP (HuFNSC01 & HuFNSC02) support the assumption that 5mC is a stable mark for exon transcription during development.   
* 5mC exon marking is established between neurospheres. _Needs further validation against H1_.   

```{r 5mC_cortexge}
gt <- ggplot_gtable(ggplot_build(WGBS_figure_HuFNSC02 + ggtitle("WGBS HuFNSC02 cortex vs GE"))) 
gt$heights[[4]] <- unit(3.5, "null") 
grid.newpage()
grid.draw(gt) 
grid.text("Average DNA methylation", x = unit(0.015, "npc"), y = unit(0.65, "npc"), rot = 90)
grid.text("CpG density", x = unit(0.015, "npc"), y = unit(0.15, "npc"), rot = 90)

(MeDIP_figure_HuFNSC02 + ggtitle("MeDIP HuFNSC02 cortex vs GE"))

gt <- ggplot_gtable(ggplot_build(WGBS_figure_HuFNSC04 + ggtitle("WGBS HuFNSC04 cortex vs GE"))) 
gt$heights[[4]] <- unit(3.5, "null") 
grid.newpage()
grid.draw(gt) 
grid.text("Average DNA methylation", x = unit(0.015, "npc"), y = unit(0.65, "npc"), rot = 90)
grid.text("CpG density", x = unit(0.015, "npc"), y = unit(0.15, "npc"), rot = 90)

(MeDIP_figure_HuFNSC01 + ggtitle("MeDIP HuFNSC01 cortex vs GE"))
```

### H3K36me3 in exon bodies
* H3K36me3 in expressed in both / not expressed exons shows no significant differences between HuFNSC01 and HuFNSC02.    
* H3K36me3 in cassette exons in HuFNSC01 are enriched in GE compared to cortex. However, it is not reproduced in HuFNSC02, where there is no significant differences between cortex and GE. _Not sure what to make of this. Are there any potential bias?_    

```{r H3K36me3_cortexge}
(H3K36me3_figure_HuFNSC01 + ggtitle("H3K36me3 HuFNSC01 cortex vs GE"))

(H3K36me3_figure_HuFNSC02 + ggtitle("H3K36me3 HuFNSC02 cortex vs GE"))
```


## Between monozygotic twins 
### DNA methylation at exon boundaries 
* 5mC for cassette exons between HuFNSC01 and HuFNSC02 shows no significant differences.   
* There are significant differences in 5mC between HuFNSC01 and HuFNSC02 specific exons in all three cell types.   
* In brain, 5mC in HuFNSC02 specific exons are closer to expressed in both, and HuFNSC01 specific exons are closer to not expressed exons. However, we observe the opposite trend in cortex and GE. _Is this a reflection of developmental stages differences between MZ twins? Are the opposite trends between brain and cortex/GE because of cell culture? Or could all these differences be technical bias / noise?_    

```{r 5mC_MZ}
(MeDIP_figure_brain + ggtitle("MeDIP HuFNSC01 vs HuFNSC02 brain"))

(MeDIP_figure_cortex + ggtitle("MeDIP HuFNSC01 vs HuFNSC02 cortex"))

(MeDIP_figure_ge + ggtitle("MeDIP HuFNSC01 vs HuFNSC02 GE"))
```

### H3K36me3 in exon bodies
* There are some differences between HuFNSC01 and HuFNSC02 in expressed in both and not expressed exons, less so in GE.   
* For cassette exons, HuFNSC02 have enriched H3K36me3 compared to HuFNSC01 in both brain and cortex, but __not__ in GE. _Could this be related to the asymmetry we observed in DNA methylation in brain and cortex?_   

```{r H3K36me3_MZ}
(H3K36me3_figure_brain + ggtitle("H3K36m3 HuFNSC01 vs HuFNSC02 brain"))

(H3K36me3_figure_cortex + ggtitle("H3K36m3 HuFNSC01 vs HuFNSC02 cortex"))

(H3K36me3_figure_ge + ggtitle("H3K36m3 HuFNSC01 vs HuFNSC02 GE"))
```






